<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Maeva Techer" />

<meta name="date" content="2022-11-01" />

<title>Overlap of the DEG detected by DEseq2 and edgeR</title>

<script src="site_libs/header-attrs-2.17/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Locust Phase Transition RNA-seq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Bioinformatics code
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-header">Sequence data pre-processing</li>
    <li>
      <a href="workenvs-prep.html">Preparation for the analysis</a>
    </li>
    <li>
      <a href="seqdata-prep.html">Download/obtain sequencing data</a>
    </li>
    <li>
      <a href="seqdata-qc.html">Data quality control</a>
    </li>
    <li class="dropdown-header">Genome alignment and gene quantification</li>
    <li>
      <a href="map-refseq.html">Mapping reads to Schistocerca RefSeq</a>
    </li>
    <li class="dropdown-header">Data post-processing and stats</li>
    <li>
      <a href="bam-stats.html">Indexing and stats on bam files</a>
    </li>
    <li>
      <a href="deseq-workflow.html">DEG-DEseq2 workflow</a>
    </li>
    <li>
      <a href="edgeR-workflow.html">DEG-EdgeR workflow</a>
    </li>
    <li>
      <a href="deg-overlap.html">DEseq2-EdgeR overlap</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Analysis &amp; Results
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-header">Mapping success</li>
    <li>
      <a href="map-star.html">Using STAR</a>
    </li>
    <li>
      <a href="map-hisat2.html">Using HISAT2</a>
    </li>
    <li class="dropdown-header">Differential Gene Expression</li>
    <li>
      <a href="deg-tissue.html">DEG per tissue</a>
    </li>
  </ul>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/MaevaTecher/locust-phase-transition-RNAseq">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Overlap of the DEG detected by DEseq2 and
edgeR</h1>
<h4 class="author">Maeva Techer</h4>
<h4 class="date">2022-11-01</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-11-01
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong>
<code>locust-phase-transition-RNAseq/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20221025code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20221025)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20221025code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20221025)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it
difficult for you and others to run your code on a different machine.
Change the absolute path(s) below to the suggested relative path(s) to
make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
/Users/alphamanae/Documents/GitHub/locust-phase-transition-RNAseq/data/piceifrons/edgeR_SPICE_HEAD
</td>
<td style="text-align:left;">
data/piceifrons/edgeR_SPICE_HEAD
</td>
</tr>
<tr>
<td style="text-align:left;">
/Users/alphamanae/Documents/GitHub/locust-phase-transition-RNAseq/data/piceifrons/DEseq2_SPICE_HEAD
</td>
<td style="text-align:left;">
data/piceifrons/DEseq2_SPICE_HEAD
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomMaevaTecherlocustphasetransitionRNAseqtree447a2e12d654a007c2fa953fc6db5e513b322cf6targetblank447a2e1a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/MaevaTecher/locust-phase-transition-RNAseq/tree/447a2e12d654a007c2fa953fc6db5e513b322cf6" target="_blank">447a2e1</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomMaevaTecherlocustphasetransitionRNAseqtree447a2e12d654a007c2fa953fc6db5e513b322cf6targetblank447a2e1a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/MaevaTecher/locust-phase-transition-RNAseq/tree/447a2e12d654a007c2fa953fc6db5e513b322cf6" target="_blank">447a2e1</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    analysis/.DS_Store
    Ignored:    data/.DS_Store
    Ignored:    data/americana/.DS_Store
    Ignored:    data/americana/STAR_counts_4thcol/.DS_Store
    Ignored:    data/cancellata/.DS_Store
    Ignored:    data/cancellata/STAR_counts_4thcol/.DS_Store
    Ignored:    data/cubense/.DS_Store
    Ignored:    data/cubense/STAR_counts_4thcol/.DS_Store
    Ignored:    data/gregaria/.DS_Store
    Ignored:    data/gregaria/STAR_counts_4thcol/.DS_Store
    Ignored:    data/gregaria/list/.DS_Store
    Ignored:    data/metadata/.DS_Store
    Ignored:    data/nitens/.DS_Store
    Ignored:    data/nitens/STAR_counts_4thcol/.DS_Store
    Ignored:    data/piceifrons/.DS_Store
    Ignored:    data/piceifrons/DEseq2_SPICE_HEAD/.DS_Store
    Ignored:    data/piceifrons/STAR_counts_4thcol/.DS_Store
    Ignored:    data/piceifrons/edgeR_SPICE_HEAD/.DS_Store
    Ignored:    data/piceifrons/list/.DS_Store

Untracked files:
    Untracked:  data/piceifrons/DE-genes_strict_[SPICE_HEAD]_[238_genes].txt
    Untracked:  data/piceifrons/DEseq2_SPICE_HEAD/DE-genes_[SPICE_HEAD_DEseq2]_[378_genes].txt
    Untracked:  data/piceifrons/edgeR_SPICE_HEAD/DE-genes_[SPICE_HEAD_edgeR]_[342_genes].txt

Unstaged changes:
    Deleted:    data/piceifrons/venn_diagramm.png

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/deg-overlap.Rmd</code>) and HTML
(<code>docs/deg-overlap.html</code>) files. If you’ve configured a
remote Git repository (see <code>?wflow_git_remote</code>), click on the
hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MaevaTecher/locust-phase-transition-RNAseq/blob/447a2e12d654a007c2fa953fc6db5e513b322cf6/analysis/deg-overlap.Rmd" target="_blank">447a2e1</a>
</td>
<td>
MaevaTecher
</td>
<td>
2022-11-01
</td>
<td>
workflowr::wflow_publish("../../analysis/deg-overlap.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MaevaTecher/locust-phase-transition-RNAseq/d1a755e4bba14faebb66ab56202b5aa8e55ab409/docs/deg-overlap.html" target="_blank">d1a755e</a>
</td>
<td>
MaevaTecher
</td>
<td>
2022-11-01
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MaevaTecher/locust-phase-transition-RNAseq/blob/4743cf7eac882ad1b0aa92e1ab1b8bee37fa8d8c/analysis/deg-overlap.Rmd" target="_blank">4743cf7</a>
</td>
<td>
MaevaTecher
</td>
<td>
2022-11-01
</td>
<td>
workflowr::wflow_publish("../../analysis/deg-overlap.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MaevaTecher/locust-phase-transition-RNAseq/71b199d0305e1e9e3f87c4a8a72479bee3f9b3ae/docs/deg-overlap.html" target="_blank">71b199d</a>
</td>
<td>
MaevaTecher
</td>
<td>
2022-11-01
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MaevaTecher/locust-phase-transition-RNAseq/blob/d997db04d9a974787e1da0ef5cfd89b28d988182/analysis/deg-overlap.Rmd" target="_blank">d997db0</a>
</td>
<td>
MaevaTecher
</td>
<td>
2022-11-01
</td>
<td>
workflowr::wflow_publish("../../analysis/deg-overlap.Rmd")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MaevaTecher/locust-phase-transition-RNAseq/1fa8b999192f9d5432c6632954c26e8b5cf61967/docs/deg-overlap.html" target="_blank">1fa8b99</a>
</td>
<td>
MaevaTecher
</td>
<td>
2022-11-01
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MaevaTecher/locust-phase-transition-RNAseq/40deb86a54d363f0d23409553dacd6a0217c6696/docs/deg-overlap.html" target="_blank">40deb86</a>
</td>
<td>
MaevaTecher
</td>
<td>
2022-11-01
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MaevaTecher/locust-phase-transition-RNAseq/83b00b823551dd8f76fd5b753dc812e4ce1cb294/docs/deg-overlap.html" target="_blank">83b00b8</a>
</td>
<td>
MaevaTecher
</td>
<td>
2022-11-01
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MaevaTecher/locust-phase-transition-RNAseq/blob/c54652554d66860d9bdfb59c0c4b8025f132f54e/analysis/deg-overlap.Rmd" target="_blank">c546525</a>
</td>
<td>
MaevaTecher
</td>
<td>
2022-11-01
</td>
<td>
workflowr::wflow_publish(c("../../analysis/deg-overlap.Rmd",
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MaevaTecher/locust-phase-transition-RNAseq/05f27e6e536e193fe3b1a228b9b03c443b6ee3bf/docs/deg-overlap.html" target="_blank">05f27e6</a>
</td>
<td>
MaevaTecher
</td>
<td>
2022-11-01
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MaevaTecher/locust-phase-transition-RNAseq/blob/25a2ba3f023fa73744c420d87d31aae63e8799e1/analysis/deg-overlap.Rmd" target="_blank">25a2ba3</a>
</td>
<td>
MaevaTecher
</td>
<td>
2022-11-01
</td>
<td>
workflowr::wflow_publish(c("../../analysis/_site.yml",
"../../analysis/edgeR-workflow.Rmd",
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="load-required-r-libraries" class="section level1">
<h1>Load required R libraries</h1>
<pre class="r"><code>#(install first from CRAN or Bioconductor)
library(&quot;knitr&quot;)
library(&quot;rmdformats&quot;)
library(&quot;tidyverse&quot;)
library(&quot;DT&quot;)  # for making interactive search table
library(&quot;plotly&quot;) # for interactive plots
library(&quot;ggthemes&quot;) # for theme_calc
library(&quot;reshape2&quot;)
library(&quot;DESeq2&quot;)
library(&quot;data.table&quot;)
library(&quot;apeglm&quot;)
library(&quot;ggpubr&quot;)
library(&quot;ggplot2&quot;)
library(&quot;ggrepel&quot;)
library(&quot;EnhancedVolcano&quot;)
library(&quot;SARTools&quot;)
library(&quot;pheatmap&quot;)
library(&quot;VennDiagram&quot;)

## Global options
options(max.print=&quot;10000&quot;)
knitr::opts_chunk$set(
    echo = TRUE,
    message = FALSE,
    warning = FALSE,
    cache = FALSE,
    comment = FALSE,
    prompt = FALSE,
    tidy = TRUE
)
opts_knit$set(width=75)</code></pre>
</div>
<div id="import-the-final-files-from-deseq2-and-edger"
class="section level1">
<h1>Import the final files from DEseq2 and edgeR</h1>
<pre class="r"><code>projectName &lt;- &quot;SPICE_HEAD&quot;  # name of the project
author &lt;- &quot;Maeva TECHER&quot;  # author of the statistical analysis/report

# Import output from edgeR/DEseq2
workDir_edgeR &lt;- &quot;/Users/alphamanae/Documents/GitHub/locust-phase-transition-RNAseq/data/piceifrons/edgeR_SPICE_HEAD&quot;
setwd(workDir_edgeR)
table_edgeR &lt;- read.table(&quot;tables/CrowdedvsIsolated.complete.txt&quot;, header = TRUE)

workDir_DEseq2 &lt;- &quot;/Users/alphamanae/Documents/GitHub/locust-phase-transition-RNAseq/data/piceifrons/DEseq2_SPICE_HEAD&quot;
setwd(workDir_DEseq2)
table_DEseq2 &lt;- read.table(&quot;tables/CrowdedvsIsolated.complete.txt&quot;, header = TRUE)

tresh_logfold &lt;- 1
tresh_padj &lt;- 0.05

# Select DE genes
table_edgeR_DE &lt;- subset(table_edgeR, table_edgeR$padj &lt;= tresh_padj &amp; (table_edgeR$log2FoldChange &gt;=
    tresh_logfold | table_edgeR$log2FoldChange &lt;= -tresh_logfold), select = Id:trended.dispersion)

table_DEseq2_DE &lt;- subset(table_DEseq2, table_DEseq2$padj &lt;= tresh_padj &amp; (table_DEseq2$log2FoldChange &gt;=
    tresh_logfold | table_DEseq2$log2FoldChange &lt;= -tresh_logfold), select = Id:maxCooks)

# Write output file
setwd(workDir_edgeR)
Filename &lt;- paste(&quot;DE-genes_[&quot;, projectName, &quot;_edgeR]_[&quot;, length(table_edgeR_DE[[1]]),
    &quot;_genes].txt&quot;, sep = &quot;&quot;)
write.table(table_edgeR_DE, file = Filename, row.names = FALSE, quote = FALSE, sep = &quot;\t&quot;)

setwd(workDir_DEseq2)
Filename &lt;- paste(&quot;DE-genes_[&quot;, projectName, &quot;_DEseq2]_[&quot;, length(table_DEseq2_DE[[1]]),
    &quot;_genes].txt&quot;, sep = &quot;&quot;)
write.table(table_DEseq2_DE, file = Filename, row.names = FALSE, quote = FALSE, sep = &quot;\t&quot;)</code></pre>
</div>
<div id="calculate-similarities-between-deseq2-and-edger"
class="section level1">
<h1>Calculate similarities between DEseq2 and edgeR</h1>
<pre class="r"><code>table_edgeR_DE$Coder &lt;- &quot;edgeR&quot;
table_edgeR_DE  &lt;- table_edgeR_DE[c(&quot;Coder&quot;, &quot;Id&quot;)]
table_DEseq2_DE$Coder &lt;- &quot;DEseq2&quot;
table_DEseq2_DE  &lt;- table_DEseq2_DE[c(&quot;Coder&quot;, &quot;Id&quot;)]
table5 &lt;- rbind(table_edgeR_DE, table_DEseq2_DE)

dupsBetweenGroups &lt;- function (df, idcol) {
    datacols &lt;- setdiff(names(df), idcol)
    sortorder &lt;- do.call(order, df)
    df &lt;- df[sortorder,]
    dupWithin &lt;- duplicated(df)
    dupBetween = rep(NA, nrow(df))
    dupBetween[!dupWithin] &lt;- duplicated(df[!dupWithin,datacols])
    dupBetween[!dupWithin] &lt;- duplicated(df[!dupWithin,datacols], fromLast=TRUE) | dupBetween[!dupWithin]
    goodIdx &lt;- !is.na(dupBetween)
    goodVals &lt;- c(NA, dupBetween[goodIdx])
    fillIdx &lt;- cumsum(goodIdx)+1
    dupBetween &lt;- goodVals[fillIdx]
    dupBetween[sortorder] &lt;- dupBetween
    return(dupBetween)}
dupRows &lt;- dupsBetweenGroups(table5, &quot;Coder&quot;)
table5 &lt;- cbind(table5, dup=dupRows)
table5 &lt;- subset(table5, Coder==&quot;edgeR&quot; &amp; dup==TRUE, select=-Coder)
setwd(&#39;..&#39;)
Filename &lt;- paste(&quot;DE-genes_strict_[&quot;, projectName, &quot;]_[&quot;, length(table5[[1]]), &quot;_genes].txt&quot;, sep = &quot;&quot;)
write.table(table5, file = Filename, row.names = FALSE, quote = FALSE, sep=&quot;\t&quot;)

#Print result statistics
cat(&quot;Amongst a total of&quot;, length(table_edgeR[[1]]), &quot;genes,
DEseq2 found&quot;, length(table_DEseq2_DE[[1]]), &quot;differentially expressed genes. 
EdgeR found&quot;,  length(table_edgeR_DE[[1]]), &quot;differentially expressed genes.
A total of&quot;, length(table5[[1]]), &quot;genes was found by both programs.&quot;)</code></pre>
<pre><code>FALSE Amongst a total of 28731 genes,
FALSE DEseq2 found 378 differentially expressed genes. 
FALSE EdgeR found 342 differentially expressed genes.
FALSE A total of 238 genes was found by both programs.</code></pre>
<pre class="r"><code>test_a_vector &lt;- as.vector(unlist(table_DEseq2_DE$Id))
test_b_vector &lt;- as.vector(unlist(table_edgeR_DE$Id))

library(&quot;RColorBrewer&quot;)
myCol &lt;- brewer.pal(3, &quot;Pastel2&quot;)

venn.plot &lt;- venn.diagram(
        x = list(test_a_vector, test_b_vector),
        category.names = c(&quot;DEseq2&quot; , &quot;EdgeR&quot;),
        filename = &#39;venn_diagramm.png&#39;,
        output=TRUE,
        
        # Output features
        imagetype=&quot;png&quot; ,
        resolution = 300,
        compression = &quot;lzw&quot;,
        
        # Circles
        lwd = 2,
        lty = &#39;blank&#39;,
        fill = c(&quot;red&quot;, &quot;blue&quot;),
        
        # Numbers
        cex = 3,
        fontface = &quot;bold&quot;,
        fontfamily = &quot;sans&quot;,
        
        # Set names
        cat.cex = 1.5,
        cat.fontface = &quot;bold&quot;,
        cat.default.pos = &quot;outer&quot;,
        cat.fontfamily = &quot;sans&quot;
)</code></pre>
<p><img src="assets/venn_diagramm.png" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>FALSE R version 4.2.1 (2022-06-23)
FALSE Platform: x86_64-apple-darwin17.0 (64-bit)
FALSE Running under: macOS Big Sur ... 10.16
FALSE 
FALSE Matrix products: default
FALSE BLAS:   /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRblas.0.dylib
FALSE LAPACK: /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRlapack.dylib
FALSE 
FALSE locale:
FALSE [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
FALSE 
FALSE attached base packages:
FALSE [1] grid      stats4    stats     graphics  grDevices utils     datasets 
FALSE [8] methods   base     
FALSE 
FALSE other attached packages:
FALSE  [1] RColorBrewer_1.1-3          VennDiagram_1.7.3          
FALSE  [3] futile.logger_1.4.3         pheatmap_1.0.12            
FALSE  [5] SARTools_1.8.1              kableExtra_1.3.4           
FALSE  [7] edgeR_3.38.4                limma_3.52.4               
FALSE  [9] ashr_2.2-54                 EnhancedVolcano_1.14.0     
FALSE [11] ggrepel_0.9.1               ggpubr_0.4.0               
FALSE [13] apeglm_1.18.0               data.table_1.14.4          
FALSE [15] DESeq2_1.36.0               SummarizedExperiment_1.26.1
FALSE [17] Biobase_2.56.0              MatrixGenerics_1.8.1       
FALSE [19] matrixStats_0.62.0          GenomicRanges_1.48.0       
FALSE [21] GenomeInfoDb_1.32.4         IRanges_2.31.2             
FALSE [23] S4Vectors_0.34.0            BiocGenerics_0.42.0        
FALSE [25] reshape2_1.4.4              ggthemes_4.2.4             
FALSE [27] plotly_4.10.0               DT_0.26                    
FALSE [29] forcats_0.5.2               stringr_1.4.1              
FALSE [31] dplyr_1.0.10                purrr_0.3.5                
FALSE [33] readr_2.1.3                 tidyr_1.2.1                
FALSE [35] tibble_3.1.8                ggplot2_3.3.6              
FALSE [37] tidyverse_1.3.2             rmdformats_1.0.4           
FALSE [39] knitr_1.40                 
FALSE 
FALSE loaded via a namespace (and not attached):
FALSE   [1] readxl_1.4.1           backports_1.4.1        workflowr_1.7.0       
FALSE   [4] systemfonts_1.0.4      plyr_1.8.7             lazyeval_0.2.2        
FALSE   [7] splines_4.2.1          BiocParallel_1.30.4    digest_0.6.30         
FALSE  [10] invgamma_1.1           htmltools_0.5.3        SQUAREM_2021.1        
FALSE  [13] fansi_1.0.3            magrittr_2.0.3         memoise_2.0.1         
FALSE  [16] googlesheets4_1.0.1    tzdb_0.3.0             Biostrings_2.64.1     
FALSE  [19] annotate_1.74.0        modelr_0.1.9           svglite_2.1.0         
FALSE  [22] bdsmatrix_1.3-6        colorspace_2.0-3       blob_1.2.3            
FALSE  [25] rvest_1.0.3            haven_2.5.1            xfun_0.34             
FALSE  [28] crayon_1.5.2           RCurl_1.98-1.9         jsonlite_1.8.3        
FALSE  [31] genefilter_1.78.0      survival_3.4-0         glue_1.6.2            
FALSE  [34] gtable_0.3.1           gargle_1.2.1           zlibbioc_1.42.0       
FALSE  [37] XVector_0.36.0         webshot_0.5.4          DelayedArray_0.22.0   
FALSE  [40] car_3.1-1              abind_1.4-5            scales_1.2.1          
FALSE  [43] futile.options_1.0.1   mvtnorm_1.1-3          GGally_2.1.2          
FALSE  [46] DBI_1.1.3              rstatix_0.7.0          Rcpp_1.0.9            
FALSE  [49] viridisLite_0.4.1      xtable_1.8-4           emdbook_1.3.12        
FALSE  [52] bit_4.0.4              truncnorm_1.0-8        htmlwidgets_1.5.4     
FALSE  [55] httr_1.4.4             ellipsis_0.3.2         reshape_0.8.9         
FALSE  [58] pkgconfig_2.0.3        XML_3.99-0.11          sass_0.4.2            
FALSE  [61] dbplyr_2.2.1           locfit_1.5-9.6         utf8_1.2.2            
FALSE  [64] tidyselect_1.2.0       rlang_1.0.6            later_1.3.0           
FALSE  [67] AnnotationDbi_1.58.0   munsell_0.5.0          cellranger_1.1.0      
FALSE  [70] tools_4.2.1            cachem_1.0.6           cli_3.4.1             
FALSE  [73] generics_0.1.3         RSQLite_2.2.18         broom_1.0.1           
FALSE  [76] ggdendro_0.1.23        evaluate_0.17          fastmap_1.1.0         
FALSE  [79] yaml_2.3.6             bit64_4.0.5            fs_1.5.2              
FALSE  [82] KEGGREST_1.36.3        whisker_0.4            formatR_1.12          
FALSE  [85] xml2_1.3.3             compiler_4.2.1         rstudioapi_0.14       
FALSE  [88] png_0.1-7              ggsignif_0.6.4         reprex_2.0.2          
FALSE  [91] geneplotter_1.74.0     bslib_0.4.0            stringi_1.7.8         
FALSE  [94] highr_0.9              lattice_0.20-45        Matrix_1.5-1          
FALSE  [97] vctrs_0.5.0            pillar_1.8.1           lifecycle_1.0.3       
FALSE [100] jquerylib_0.1.4        irlba_2.3.5.1          bitops_1.0-7          
FALSE [103] httpuv_1.6.6           R6_2.5.1               bookdown_0.29         
FALSE [106] promises_1.2.0.1       gridExtra_2.3          codetools_0.2-18      
FALSE [109] lambda.r_1.2.4         MASS_7.3-58.1          assertthat_0.2.1      
FALSE [112] rprojroot_2.0.3        withr_2.5.0            GenomeInfoDbData_1.2.8
FALSE [115] parallel_4.2.1         hms_1.1.2              coda_0.19-4           
FALSE [118] rmarkdown_2.17         carData_3.0-5          googledrive_2.0.0     
FALSE [121] git2r_0.30.1           mixsqp_0.3-43          bbmle_1.0.25          
FALSE [124] numDeriv_2016.8-1.1    lubridate_1.8.0</code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
