<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Maeva Techer" />

<meta name="date" content="2022-10-31" />

<title>Differential Gene Expression Workflow</title>

<script src="site_libs/header-attrs-2.17/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Locust Phase Transition RNA-seq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Bioinformatics code
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-header">Sequence data pre-processing</li>
    <li>
      <a href="workenvs-prep.html">Preparation for the analysis</a>
    </li>
    <li>
      <a href="seqdata-prep.html">Download/obtain sequencing data</a>
    </li>
    <li>
      <a href="seqdata-qc.html">Data quality control</a>
    </li>
    <li class="dropdown-header">Genome alignment and gene quantification</li>
    <li>
      <a href="map-refseq.html">Mapping reads to Schistocerca RefSeq</a>
    </li>
    <li class="dropdown-header">Data post-processing and stats</li>
    <li>
      <a href="bam-stats.html">Indexing and stats on bam files</a>
    </li>
    <li>
      <a href="deg-workflow.html">DEG workflow</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Analysis &amp; Results
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-header">Mapping success</li>
    <li>
      <a href="map-star.html">Using STAR</a>
    </li>
    <li>
      <a href="map-hisat2.html">Using HISAT2</a>
    </li>
    <li class="dropdown-header">Differential Gene Expression</li>
    <li>
      <a href="deg-tissue.html">DEG per tissue</a>
    </li>
  </ul>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/MaevaTecher/locust-phase-transition-RNAseq">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Differential Gene Expression Workflow</h1>
<h4 class="author">Maeva Techer</h4>
<h4 class="date">2022-10-31</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-10-31
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong>
<code>locust-phase-transition-RNAseq/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20221025code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20221025)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20221025code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20221025)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomMaevaTecherlocustphasetransitionRNAseqtree2ccb4bc5e57b273db98e89c4b87c580ee510cd67targetblank2ccb4bca">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/MaevaTecher/locust-phase-transition-RNAseq/tree/2ccb4bc5e57b273db98e89c4b87c580ee510cd67" target="_blank">2ccb4bc</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomMaevaTecherlocustphasetransitionRNAseqtree2ccb4bc5e57b273db98e89c4b87c580ee510cd67targetblank2ccb4bca"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/MaevaTecher/locust-phase-transition-RNAseq/tree/2ccb4bc5e57b273db98e89c4b87c580ee510cd67" target="_blank">2ccb4bc</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    analysis/.DS_Store
    Ignored:    data/.DS_Store
    Ignored:    data/americana/.DS_Store
    Ignored:    data/americana/STAR_counts_4thcol/.DS_Store
    Ignored:    data/cancellata/.DS_Store
    Ignored:    data/cancellata/STAR_counts_4thcol/.DS_Store
    Ignored:    data/cubense/.DS_Store
    Ignored:    data/cubense/STAR_counts_4thcol/.DS_Store
    Ignored:    data/gregaria/.DS_Store
    Ignored:    data/gregaria/STAR_counts_4thcol/.DS_Store
    Ignored:    data/gregaria/list/.DS_Store
    Ignored:    data/metadata/.DS_Store
    Ignored:    data/nitens/.DS_Store
    Ignored:    data/nitens/STAR_counts_4thcol/.DS_Store
    Ignored:    data/piceifrons/.DS_Store
    Ignored:    data/piceifrons/STAR_counts_4thcol/.DS_Store
    Ignored:    data/piceifrons/list/.DS_Store

Unstaged changes:
    Deleted:    analysis/gene-quant.Rmd
    Deleted:    analysis/own-refseq.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/deg-workflow.Rmd</code>) and HTML
(<code>docs/deg-workflow.html</code>) files. If you’ve configured a
remote Git repository (see <code>?wflow_git_remote</code>), click on the
hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MaevaTecher/locust-phase-transition-RNAseq/blob/2ccb4bc5e57b273db98e89c4b87c580ee510cd67/analysis/deg-workflow.Rmd" target="_blank">2ccb4bc</a>
</td>
<td>
MaevaTecher
</td>
<td>
2022-10-31
</td>
<td>
wflow_publish(c("analysis/_site.yml",
"data/gregaria/list/BrainPilot.txt",
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="load-required-r-libraries" class="section level1">
<h1>Load required R libraries</h1>
<pre class="r"><code>#(install first from CRAN or Bioconductor)
library(&quot;knitr&quot;)
library(&quot;rmdformats&quot;)
library(&quot;tidyverse&quot;)
library(&quot;DT&quot;)  # for making interactive search table
library(&quot;plotly&quot;) # for interactive plots
library(&quot;ggthemes&quot;) # for theme_calc
library(&quot;reshape2&quot;)
library(&quot;DESeq2&quot;)
library(&quot;data.table&quot;)
library(&quot;apeglm&quot;)
library(&quot;ggpubr&quot;)
library(&quot;ggplot2&quot;)
library(&quot;ggrepel&quot;)
library(&quot;EnhancedVolcano&quot;)

## Global options
options(max.print=&quot;10000&quot;)
knitr::opts_chunk$set(
    echo = TRUE,
    message = FALSE,
    warning = FALSE,
    cache = FALSE,
    comment = FALSE,
    prompt = FALSE,
    tidy = TRUE
)
opts_knit$set(width=75)</code></pre>
<p>Here we present the workflow example with the head data from <em>S.
piceifrons</em></p>
<p>For the analysis of differentially expressed genes, we will follow
some guidelines from an online RNA course <a
href="https://biocorecrg.github.io/RNAseq_course_2019/differential_expression.html">tutorial</a>
that uses either <code>DESeq2</code> or <code>edgeR</code> on STAR
output.</p>
<p>DESeq2 tests for differential expression using negative binomial
generalized linear models. DESeq2 (as edgeR) is based on the hypothesis
that most genes are not differentially expressed. The package takes as
an input raw counts (i.e. non normalized counts): the DESeq2 model
internally corrects for library size, so giving as an input normalized
count would be incorrect.</p>
</div>
<div id="dge-analysis-with-star-input" class="section level1">
<h1>DGE analysis with STAR input</h1>
<div id="preparing-input-files" class="section level2">
<h2>Preparing input files</h2>
<div id="raw-count-matrices" class="section level3">
<h3>Raw count matrices</h3>
<p>We generated this in the precedent section.</p>
</div>
<div id="transcript-to-gene-annotation-file" class="section level3">
<h3>Transcript-to-gene annotation file</h3>
<p>Below is the example with <em>S. gregaria</em></p>
<pre><code># Download annotation and place it into the folder refgenomes
wget https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/023/897/955/GCF_023897955.1_iqSchGreg1.2/GCF_023897955.1_iqSchGreg1.2_genomic.gtf.gz

# first column is the transcript ID, second column is the gene ID, third column is the gene symbol
zcat GCF_023897955.1_iqSchGreg1.2_genomic.gtf.gz | awk -F &quot;\t&quot; &#39;BEGIN{OFS=&quot;\t&quot;}{if($3==&quot;transcript&quot;){split($9, a, &quot;\&quot;&quot;); print a[4],a[2],a[8]}}&#39; &gt; tx2gene.gregaria.csv</code></pre>
</div>
<div id="sample-sheet" class="section level3">
<h3>Sample sheet</h3>
<p>DESeq2 needs a sample sheet that describes the samples
characteristics: SampleName, FileName (…counts.txt), and subsequently
anything that can be used for <strong>statistical design</strong> such
as RearingCondition, replicates, tissue, time points, etc. in the
form.</p>
<p>The design indicates how to model the samples: in the model we need
to specify what we want to measure and what we want to
<strong>control</strong>.</p>
</div>
</div>
</div>
<div id="differential-expression-analysis" class="section level1">
<h1>Differential expression analysis</h1>
<div id="import-input-count-data" class="section level2">
<h2>Import input count data</h2>
<p>We start by reading the sample sheet.</p>
<pre class="r"><code>## import the sample sheet that indicates Rearing Conditions and Tissue origins
sampletable &lt;- fread(&quot;data/piceifrons/list/HeadSPICE.txt&quot;)
# add the sample names as row names (it is needed for some of the DESeq
# functions)
rownames(sampletable) &lt;- sampletable$SampleName

# Make sure discriminant variables are factor
sampletable$RearingCondition &lt;- as.factor(sampletable$RearingCondition)
sampletable$Tissue &lt;- as.factor(sampletable$Tissue)
dim(sampletable)</code></pre>
<pre><code>FALSE [1] 10  4</code></pre>
<p>Then we obtain the output from <code>STAR GeneCount</code> and import
here individually using the sampletable as a reference to fetch them. We
also filter the lowly expressed genes to avoid noisy data.</p>
<pre class="r"><code># Read in the two-column data.frame linking transcript id (column 1) to gene id
# (column 2)
tx2gene &lt;- read.table(&quot;data/piceifrons/list/tx2gene.piceifrons.csv&quot;, sep = &quot;\t&quot;,
    header = F)

satoshi &lt;- DESeqDataSetFromHTSeqCount(sampleTable = sampletable, directory = &quot;data/piceifrons/STAR_counts_4thcol&quot;,
    design = ~RearingCondition)

satoshi</code></pre>
<pre><code>FALSE class: DESeqDataSet 
FALSE dim: 28731 10 
FALSE metadata(1): version
FALSE assays(1): counts
FALSE rownames(28731): LOC124794980 LOC124795035 ... LOC124774866
FALSE   LOC124774858
FALSE rowData names(0):
FALSE colnames(10): SPICE_G_Crd_SRR11815268 SPICE_G_Crd_SRR11815269 ...
FALSE   SPICE_S_Iso_SRR11815273 SPICE_S_Iso_SRR11815274
FALSE colData names(2): Tissue RearingCondition</code></pre>
<pre class="r"><code># keep genes for which sums of raw counts across experimental samples is &gt; 5
satoshi &lt;- satoshi[rowSums(counts(satoshi)) &gt; 5, ]
nrow(satoshi)</code></pre>
<pre><code>FALSE [1] 14932</code></pre>
<pre class="r"><code># set a standard to be compared to (hatchling) ONLY IF WE HAVE A CONTROL
# satoshi$Tissue &lt;- relevel(satoshi$Tissue, ref = &#39;Whole_body&#39;)</code></pre>
</div>
<div id="run-the-model-fit-deseq2" class="section level2">
<h2>Run the model Fit DESeq2</h2>
<p>Run DESeq2 analysis using <code>DESeq</code>, which performs (1)
estimation of size factors, (2) estimation of dispersion, then (3)
Negative Binomial GLM fitting and Wald statistics. The results tables
(log2 fold changes and p-values) can be generated using the results
function (<a
href="https://ycl6.gitbook.io/guide-to-rna-seq-analysis/differential-expression-analysis/differential-gene-expression/dge-analysis-with-star-input">copied
from online chapter</a>)</p>
<pre class="r"><code># Fit the statistical model
shigeru &lt;- DESeq(satoshi)
cbind(resultsNames(shigeru))</code></pre>
<pre><code>FALSE      [,1]                                  
FALSE [1,] &quot;Intercept&quot;                           
FALSE [2,] &quot;RearingCondition_Isolated_vs_Crowded&quot;</code></pre>
<pre class="r"><code># Here we plot the adjusted p-value means corrected for multiple testing (FDR
# padj)
res_shigeru &lt;- results(shigeru)

# We only keep genes with an adjusted p-value cutoff &gt; 0.05 by changing the
# default significance cut-off sum(res_shigeru$padj &lt; 0.05, na.rm = TRUE)

brock &lt;- results(shigeru, name = &quot;RearingCondition_Isolated_vs_Crowded&quot;, alpha = 0.05)
summary(brock)</code></pre>
<pre><code>FALSE 
FALSE out of 14932 with nonzero total read count
FALSE adjusted p-value &lt; 0.05
FALSE LFC &gt; 0 (up)       : 254, 1.7%
FALSE LFC &lt; 0 (down)     : 404, 2.7%
FALSE outliers [1]       : 497, 3.3%
FALSE low counts [2]     : 1156, 7.7%
FALSE (mean count &lt; 2)
FALSE [1] see &#39;cooksCutoff&#39; argument of ?results
FALSE [2] see &#39;independentFiltering&#39; argument of ?results</code></pre>
<pre class="r"><code># Details of what is each column meaning in our final result
mcols(brock)$description</code></pre>
<pre><code>FALSE [1] &quot;mean of normalized counts for all samples&quot;                   
FALSE [2] &quot;log2 fold change (MLE): RearingCondition Isolated vs Crowded&quot;
FALSE [3] &quot;standard error: RearingCondition Isolated vs Crowded&quot;        
FALSE [4] &quot;Wald statistic: RearingCondition Isolated vs Crowded&quot;        
FALSE [5] &quot;Wald test p-value: RearingCondition Isolated vs Crowded&quot;     
FALSE [6] &quot;BH adjusted p-values&quot;</code></pre>
<pre class="r"><code>head(brock)</code></pre>
<pre><code>FALSE log2 fold change (MLE): RearingCondition Isolated vs Crowded 
FALSE Wald test p-value: RearingCondition Isolated vs Crowded 
FALSE DataFrame with 6 rows and 6 columns
FALSE               baseMean log2FoldChange     lfcSE      stat      pvalue
FALSE              &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;   &lt;numeric&gt;
FALSE LOC124796288   2.37306      -3.529621  1.337214 -2.639533 8.30204e-03
FALSE LOC124796294   1.14866      -2.301130  2.134062 -1.078286 2.80906e-01
FALSE LOC124796332   9.18581      -0.600937  0.715740 -0.839603 4.01131e-01
FALSE LOC124793320 133.76396      -1.759458  0.613514 -2.867835 4.13290e-03
FALSE LOC124712404  29.00252      -1.601387  1.002767 -1.596968          NA
FALSE LOC124798649  11.86003       5.205485  0.872809  5.964059 2.46048e-09
FALSE                     padj
FALSE                &lt;numeric&gt;
FALSE LOC124796288 1.10023e-01
FALSE LOC124796294          NA
FALSE LOC124796332 7.40311e-01
FALSE LOC124793320 7.03370e-02
FALSE LOC124712404          NA
FALSE LOC124798649 6.40640e-07</code></pre>
<p>For this data set after FDR filtering of 0.05, we have 254 genes
up-regulates and 404 genes down-regulated in crowded versus solitary
individuals.</p>
</div>
</div>
<div id="visualizing-and-exploring-the-results" class="section level1">
<h1>Visualizing and exploring the results</h1>
<div id="pca-of-the-samples" class="section level2">
<h2>PCA of the samples</h2>
<p>We transformed the data for visualization by comparing both
recommended <strong>rlog</strong> (Regularized log) or
<strong>vst</strong> (Variance Stabilizing Transformation)
transformations. Both options produce log2 scale data which has been
normalized by the DESeq2 method with respect to library size.</p>
<pre class="r"><code># Try with the vst transformation
shigeru_vsd &lt;- vst(shigeru)
shigeru_rlog &lt;- rlog(shigeru)</code></pre>
<p>Plot the PCA rlog</p>
<pre class="r"><code># Create the pca on the defined groups
pcaData &lt;- plotPCA(object = shigeru_rlog, intgroup = c(&quot;RearingCondition&quot;, &quot;Tissue&quot;),
    returnData = TRUE)

# Store the information for each axis variance in %
percentVar &lt;- round(100 * attr(pcaData, &quot;percentVar&quot;))

# Make sure that the discriminant variable are in factor for using shape and
# color later on
pcaData$RearingCondition &lt;- factor(pcaData$RearingCondition, levels = c(&quot;Crowded&quot;,
    &quot;Isolated&quot;), labels = c(&quot;crowded piceifrons&quot;, &quot;isolated piceifrons&quot;))
levels(pcaData$RearingCondition)</code></pre>
<pre><code>FALSE [1] &quot;crowded piceifrons&quot;  &quot;isolated piceifrons&quot;</code></pre>
<pre class="r"><code># pcaData$Tissue&lt;-factor(pcaData1$Tissue,levels=c(&#39;Whole_body&#39;,&#39;Optical_lobes&#39;),
# labels=c(&#39;Hatchling&#39;, &#39;OLB&#39;)) levels(pcaData$Tissue)


ggplot(pcaData, aes(PC1, PC2, color = RearingCondition)) + geom_point(size = 4) +
    xlab(paste0(&quot;PC1: &quot;, percentVar[1], &quot;% variance&quot;)) + ylab(paste0(&quot;PC2: &quot;, percentVar[2],
    &quot;% variance&quot;)) + scale_color_manual(values = c(&quot;blue&quot;, &quot;red&quot;)) + geom_text_repel(aes(label = name),
    nudge_x = -1, nudge_y = 0.2, size = 3) + coord_fixed() + theme_bw() + theme(legend.title = element_blank()) +
    theme(legend.text = element_text(face = &quot;bold&quot;, size = 15)) + theme(axis.text = element_text(size = 15)) +
    theme(axis.title = element_text(size = 16)) + ggtitle(&quot;PCA on S. piceifrons head tissues&quot;,
    subtitle = &quot;rlog transformation&quot;) + xlab(paste0(&quot;PC1: &quot;, percentVar[1], &quot;% variance&quot;)) +
    ylab(paste0(&quot;PC2: &quot;, percentVar[2], &quot;% variance&quot;))</code></pre>
<p><img src="figure/deg-workflow.Rmd/DESeq2%20PCA%20rlog-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Plot the PCA vsd</p>
<pre class="r"><code># Create the pca on the defined groups
pcaData &lt;- plotPCA(object = shigeru_vsd, intgroup = c(&quot;RearingCondition&quot;, &quot;Tissue&quot;),
    returnData = TRUE)

# Store the information for each axis variance in %
percentVar &lt;- round(100 * attr(pcaData, &quot;percentVar&quot;))

# Make sure that the discriminant variable are in factor for using shape and
# color later on
pcaData$RearingCondition &lt;- factor(pcaData$RearingCondition, levels = c(&quot;Crowded&quot;,
    &quot;Isolated&quot;), labels = c(&quot;crowded piceifrons&quot;, &quot;isolated piceifrons&quot;))
levels(pcaData$RearingCondition)</code></pre>
<pre><code>FALSE [1] &quot;crowded piceifrons&quot;  &quot;isolated piceifrons&quot;</code></pre>
<pre class="r"><code># pcaData$Tissue&lt;-factor(pcaData1$Tissue,levels=c(&#39;Whole_body&#39;,&#39;Optical_lobes&#39;),
# labels=c(&#39;Hatchling&#39;, &#39;OLB&#39;)) levels(pcaData$Tissue)


ggplot(pcaData, aes(PC1, PC2, color = RearingCondition)) + geom_point(size = 4) +
    xlab(paste0(&quot;PC1: &quot;, percentVar[1], &quot;% variance&quot;)) + ylab(paste0(&quot;PC2: &quot;, percentVar[2],
    &quot;% variance&quot;)) + scale_color_manual(values = c(&quot;blue&quot;, &quot;red&quot;)) + geom_text_repel(aes(label = name),
    nudge_x = -1, nudge_y = 0.2, size = 3) + coord_fixed() + theme_bw() + theme(legend.title = element_blank()) +
    theme(legend.text = element_text(face = &quot;bold&quot;, size = 15)) + theme(axis.text = element_text(size = 15)) +
    theme(axis.title = element_text(size = 16)) + ggtitle(&quot;PCA on S. piceifrons head tissues&quot;,
    subtitle = &quot;vst transformation&quot;) + xlab(paste0(&quot;PC1: &quot;, percentVar[1], &quot;% variance&quot;)) +
    ylab(paste0(&quot;PC2: &quot;, percentVar[2], &quot;% variance&quot;))</code></pre>
<p><img src="figure/deg-workflow.Rmd/DESeq2%20PCA%20vsd-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="ma-plot" class="section level2">
<h2>MA-plot</h2>
<p>This plot allows us to show the log2 fold changes over the mean of
normalized counts for all the samples. Points will be colored in red if
the adjusted p-value is less than 0.05 and the log2 fold change is
bigger than 1. In blue, will be the reverse for the log2 fold
change.</p>
<pre class="r"><code># include the log2FoldChange shrinkage use to visualize gene ranking
de_shrink &lt;- lfcShrink(dds = shigeru, coef = &quot;RearingCondition_Isolated_vs_Crowded&quot;,
    type = &quot;apeglm&quot;)

# Ma plot
maplot &lt;- ggmaplot(de_shrink, fdr = 0.05, fc = 1, size = 1.5, palette = c(&quot;#B31B21&quot;,
    &quot;#1465AC&quot;, &quot;darkgray&quot;), genenames = as.vector(rownames(de_shrink$name)), top = 0,
    legend = &quot;top&quot;, label.select = NULL) + coord_cartesian(xlim = c(0, 18)) + scale_y_continuous(limits = c(-8,
    8)) + theme(axis.text.x = element_text(size = 16), axis.text.y = element_text(size = 15),
    axis.title.x = element_text(size = 17), axis.title.y = element_text(size = 17),
    axis.line = element_line(size = 1, colour = &quot;gray20&quot;), axis.ticks = element_line(size = 1,
        colour = &quot;gray20&quot;)) + guides(color = guide_legend(override.aes = list(size = c(3,
    3, 3)))) + theme(legend.position = c(0.7, 0.12), legend.text = element_text(size = 14,
    face = &quot;bold&quot;), legend.background = element_rect(fill = &quot;transparent&quot;)) + theme(plot.title = element_text(size = 18,
    colour = &quot;gray30&quot;, face = &quot;bold&quot;, hjust = 0.06, vjust = -5)) + labs(title = &quot;MA-plot for the shrunken log2 fold changes in the head&quot;)
maplot</code></pre>
<p><img src="figure/deg-workflow.Rmd/DESeq2%20MA%20plot%20-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>The <code>EnhancedVolcano</code> helps visualise the resulst of
differential expression analysis.</p>
<pre class="r"><code>keyvals &lt;- ifelse(res_shigeru$log2FoldChange &gt;= 1 &amp; res_shigeru$padj &lt;= 0.05, &quot;red&quot;,
    ifelse(res_shigeru$log2FoldChange &lt;= -1 &amp; res_shigeru$padj &lt;= 0.05, &quot;green&quot;,
        &quot;black&quot;))

keyvals[is.na(keyvals)] &lt;- &quot;black&quot;
names(keyvals)[keyvals == &quot;red&quot;] &lt;- &quot;Upregulated&quot;
names(keyvals)[keyvals == &quot;green&quot;] &lt;- &quot;Downregulated&quot;
names(keyvals)[keyvals == &quot;black&quot;] &lt;- &quot;NS&quot;

EnhancedVolcano(res_shigeru, lab = rownames(res_shigeru), x = &quot;log2FoldChange&quot;, y = &quot;padj&quot;,
    pCutoff = 0.05, FCcutoff = 1, pointSize = 3, labSize = 3, colAlpha = 4/5, colCustom = keyvals,
    drawConnectors = FALSE)</code></pre>
<p><img src="figure/deg-workflow.Rmd/DESeq2%20Volcano%20plot%20-1.png" width="672" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>FALSE R version 4.2.1 (2022-06-23)
FALSE Platform: x86_64-apple-darwin17.0 (64-bit)
FALSE Running under: macOS Big Sur ... 10.16
FALSE 
FALSE Matrix products: default
FALSE BLAS:   /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRblas.0.dylib
FALSE LAPACK: /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRlapack.dylib
FALSE 
FALSE locale:
FALSE [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
FALSE 
FALSE attached base packages:
FALSE [1] stats4    stats     graphics  grDevices utils     datasets  methods  
FALSE [8] base     
FALSE 
FALSE other attached packages:
FALSE  [1] EnhancedVolcano_1.14.0      ggrepel_0.9.1              
FALSE  [3] ggpubr_0.4.0                apeglm_1.18.0              
FALSE  [5] data.table_1.14.4           DESeq2_1.36.0              
FALSE  [7] SummarizedExperiment_1.26.1 Biobase_2.56.0             
FALSE  [9] MatrixGenerics_1.8.1        matrixStats_0.62.0         
FALSE [11] GenomicRanges_1.48.0        GenomeInfoDb_1.32.4        
FALSE [13] IRanges_2.31.2              S4Vectors_0.34.0           
FALSE [15] BiocGenerics_0.42.0         reshape2_1.4.4             
FALSE [17] ggthemes_4.2.4              plotly_4.10.0              
FALSE [19] DT_0.26                     forcats_0.5.2              
FALSE [21] stringr_1.4.1               dplyr_1.0.10               
FALSE [23] purrr_0.3.5                 readr_2.1.3                
FALSE [25] tidyr_1.2.1                 tibble_3.1.8               
FALSE [27] ggplot2_3.3.6               tidyverse_1.3.2            
FALSE [29] rmdformats_1.0.4            knitr_1.40                 
FALSE [31] workflowr_1.7.0            
FALSE 
FALSE loaded via a namespace (and not attached):
FALSE   [1] readxl_1.4.1           backports_1.4.1        plyr_1.8.7            
FALSE   [4] lazyeval_0.2.2         splines_4.2.1          BiocParallel_1.30.4   
FALSE   [7] digest_0.6.30          htmltools_0.5.3        fansi_1.0.3           
FALSE  [10] magrittr_2.0.3         memoise_2.0.1          googlesheets4_1.0.1   
FALSE  [13] tzdb_0.3.0             Biostrings_2.64.1      annotate_1.74.0       
FALSE  [16] modelr_0.1.9           bdsmatrix_1.3-6        colorspace_2.0-3      
FALSE  [19] blob_1.2.3             rvest_1.0.3            haven_2.5.1           
FALSE  [22] xfun_0.34              callr_3.7.2            crayon_1.5.2          
FALSE  [25] RCurl_1.98-1.9         jsonlite_1.8.3         genefilter_1.78.0     
FALSE  [28] survival_3.4-0         glue_1.6.2             gtable_0.3.1          
FALSE  [31] gargle_1.2.1           zlibbioc_1.42.0        XVector_0.36.0        
FALSE  [34] DelayedArray_0.22.0    car_3.1-1              abind_1.4-5           
FALSE  [37] scales_1.2.1           mvtnorm_1.1-3          DBI_1.1.3             
FALSE  [40] rstatix_0.7.0          Rcpp_1.0.9             viridisLite_0.4.1     
FALSE  [43] xtable_1.8-4           emdbook_1.3.12         bit_4.0.4             
FALSE  [46] htmlwidgets_1.5.4      httr_1.4.4             RColorBrewer_1.1-3    
FALSE  [49] ellipsis_0.3.2         farver_2.1.1           pkgconfig_2.0.3       
FALSE  [52] XML_3.99-0.11          sass_0.4.2             dbplyr_2.2.1          
FALSE  [55] locfit_1.5-9.6         utf8_1.2.2             labeling_0.4.2        
FALSE  [58] tidyselect_1.2.0       rlang_1.0.6            later_1.3.0           
FALSE  [61] AnnotationDbi_1.58.0   munsell_0.5.0          cellranger_1.1.0      
FALSE  [64] tools_4.2.1            cachem_1.0.6           cli_3.4.1             
FALSE  [67] generics_0.1.3         RSQLite_2.2.18         broom_1.0.1           
FALSE  [70] evaluate_0.17          fastmap_1.1.0          yaml_2.3.6            
FALSE  [73] processx_3.7.0         bit64_4.0.5            fs_1.5.2              
FALSE  [76] KEGGREST_1.36.3        whisker_0.4            formatR_1.12          
FALSE  [79] xml2_1.3.3             compiler_4.2.1         rstudioapi_0.14       
FALSE  [82] png_0.1-7              ggsignif_0.6.4         reprex_2.0.2          
FALSE  [85] geneplotter_1.74.0     bslib_0.4.0            stringi_1.7.8         
FALSE  [88] highr_0.9              ps_1.7.1               lattice_0.20-45       
FALSE  [91] Matrix_1.5-1           vctrs_0.5.0            pillar_1.8.1          
FALSE  [94] lifecycle_1.0.3        jquerylib_0.1.4        bitops_1.0-7          
FALSE  [97] httpuv_1.6.6           R6_2.5.1               bookdown_0.29         
FALSE [100] promises_1.2.0.1       codetools_0.2-18       MASS_7.3-58.1         
FALSE [103] assertthat_0.2.1       rprojroot_2.0.3        withr_2.5.0           
FALSE [106] GenomeInfoDbData_1.2.8 parallel_4.2.1         hms_1.1.2             
FALSE [109] grid_4.2.1             coda_0.19-4            rmarkdown_2.17        
FALSE [112] carData_3.0-5          googledrive_2.0.0      git2r_0.30.1          
FALSE [115] getPass_0.2-2          bbmle_1.0.25           numDeriv_2016.8-1.1   
FALSE [118] lubridate_1.8.0</code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
