<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Maeva Techer" />

<meta name="date" content="2022-11-06" />

<title>Mapping reads to Schistocerca RefSeq genome</title>

<script src="site_libs/header-attrs-2.17/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Locust Phase Transition RNA-seq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Bioinformatics code
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-header">Sequence data pre-processing</li>
    <li>
      <a href="workenvs-prep.html">Preparation for the analysis</a>
    </li>
    <li>
      <a href="seqdata-prep.html">Download/obtain sequencing data</a>
    </li>
    <li>
      <a href="seqdata-qc.html">Data quality control</a>
    </li>
    <li class="dropdown-header">Genome alignment and gene quantification</li>
    <li>
      <a href="map-refseq.html">Mapping reads to Schistocerca RefSeq</a>
    </li>
    <li class="dropdown-header">Data post-processing and stats</li>
    <li>
      <a href="bam-stats.html">Indexing and stats on bam files</a>
    </li>
    <li>
      <a href="deseq-workflow.html">DEG-DEseq2 workflow</a>
    </li>
    <li>
      <a href="edgeR-workflow.html">DEG-EdgeR workflow</a>
    </li>
    <li>
      <a href="deg-overlap.html">DEseq2-EdgeR overlap</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Analysis &amp; Results
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-header">Mapping success</li>
    <li>
      <a href="map-star.html">Using STAR</a>
    </li>
    <li>
      <a href="map-hisat2.html">Using HISAT2</a>
    </li>
    <li class="dropdown-header">Differential Gene Expression</li>
    <li>
      <a href="deg-tissue.html">DEG per tissue</a>
    </li>
  </ul>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/MaevaTecher/locust-phase-transition-RNAseq">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Mapping reads to Schistocerca RefSeq
genome</h1>
<h4 class="author">Maeva Techer</h4>
<h4 class="date">2022-11-06</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-11-06
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong>
<code>locust-phase-transition-RNAseq/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20221025code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20221025)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20221025code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20221025)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomMaevaTecherlocustphasetransitionRNAseqtree4e6386baec1e068b7f9fbdf265a87504e09b25e5targetblank4e6386ba">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/MaevaTecher/locust-phase-transition-RNAseq/tree/4e6386baec1e068b7f9fbdf265a87504e09b25e5" target="_blank">4e6386b</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomMaevaTecherlocustphasetransitionRNAseqtree4e6386baec1e068b7f9fbdf265a87504e09b25e5targetblank4e6386ba"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/MaevaTecher/locust-phase-transition-RNAseq/tree/4e6386baec1e068b7f9fbdf265a87504e09b25e5" target="_blank">4e6386b</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    analysis/.DS_Store
    Ignored:    data/.DS_Store
    Ignored:    data/americana/.DS_Store
    Ignored:    data/americana/STAR_counts_4thcol/.DS_Store
    Ignored:    data/cancellata/.DS_Store
    Ignored:    data/cancellata/STAR_counts_4thcol/.DS_Store
    Ignored:    data/cubense/.DS_Store
    Ignored:    data/cubense/STAR_counts_4thcol/.DS_Store
    Ignored:    data/gregaria/.DS_Store
    Ignored:    data/gregaria/STAR_counts_4thcol/.DS_Store
    Ignored:    data/gregaria/list/.DS_Store
    Ignored:    data/metadata/.DS_Store
    Ignored:    data/nitens/.DS_Store
    Ignored:    data/nitens/STAR_counts_4thcol/.DS_Store
    Ignored:    data/piceifrons/.DS_Store
    Ignored:    data/piceifrons/.Rhistory
    Ignored:    data/piceifrons/DEseq2_SPICE_HEAD 2/
    Ignored:    data/piceifrons/DEseq2_SPICE_HEAD/.DS_Store
    Ignored:    data/piceifrons/STAR_counts_4thcol/.DS_Store
    Ignored:    data/piceifrons/edgeR_SPICE_HEAD/.DS_Store
    Ignored:    data/piceifrons/list/.DS_Store

Unstaged changes:
    Modified:   data/metadata/RNAseq_modified_METADATA2022.xlsx
    Deleted:    data/metadata/~$RNAseq_modified_METADATA2022.xlsx

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/map-refseq.Rmd</code>) and HTML
(<code>docs/map-refseq.html</code>) files. If you’ve configured a remote
Git repository (see <code>?wflow_git_remote</code>), click on the
hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MaevaTecher/locust-phase-transition-RNAseq/blob/4e6386baec1e068b7f9fbdf265a87504e09b25e5/analysis/map-refseq.Rmd" target="_blank">4e6386b</a>
</td>
<td>
MaevaTecher
</td>
<td>
2022-11-06
</td>
<td>
workflowr::wflow_publish("analysis/map-refseq.RmD")
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MaevaTecher/locust-phase-transition-RNAseq/1fa8b999192f9d5432c6632954c26e8b5cf61967/docs/map-refseq.html" target="_blank">1fa8b99</a>
</td>
<td>
MaevaTecher
</td>
<td>
2022-11-01
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MaevaTecher/locust-phase-transition-RNAseq/40deb86a54d363f0d23409553dacd6a0217c6696/docs/map-refseq.html" target="_blank">40deb86</a>
</td>
<td>
MaevaTecher
</td>
<td>
2022-11-01
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MaevaTecher/locust-phase-transition-RNAseq/83b00b823551dd8f76fd5b753dc812e4ce1cb294/docs/map-refseq.html" target="_blank">83b00b8</a>
</td>
<td>
MaevaTecher
</td>
<td>
2022-11-01
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MaevaTecher/locust-phase-transition-RNAseq/05f27e6e536e193fe3b1a228b9b03c443b6ee3bf/docs/map-refseq.html" target="_blank">05f27e6</a>
</td>
<td>
MaevaTecher
</td>
<td>
2022-11-01
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MaevaTecher/locust-phase-transition-RNAseq/ec837789f2bf57a44726be47dc108043c8461e37/docs/map-refseq.html" target="_blank">ec83778</a>
</td>
<td>
MaevaTecher
</td>
<td>
2022-11-01
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MaevaTecher/locust-phase-transition-RNAseq/834088cdafcd35398d7a71b0c6b8b2ac92130779/docs/map-refseq.html" target="_blank">834088c</a>
</td>
<td>
MaevaTecher
</td>
<td>
2022-10-31
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MaevaTecher/locust-phase-transition-RNAseq/8481823d9015b1cc26ecb64e487d74f829d2643f/docs/map-refseq.html" target="_blank">8481823</a>
</td>
<td>
MaevaTecher
</td>
<td>
2022-10-31
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MaevaTecher/locust-phase-transition-RNAseq/ea70380071caa0d86932413a6c0120d766a79768/docs/map-refseq.html" target="_blank">ea70380</a>
</td>
<td>
MaevaTecher
</td>
<td>
2022-10-31
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MaevaTecher/locust-phase-transition-RNAseq/blob/de64b6bfd0f4a47e2f2cae212c8eb07a05a30c10/analysis/map-refseq.Rmd" target="_blank">de64b6b</a>
</td>
<td>
MaevaTecher
</td>
<td>
2022-10-31
</td>
<td>
wflow_publish(c("analysis/_site.yml", "analysis/map-refseq.Rmd",
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MaevaTecher/locust-phase-transition-RNAseq/8b34f19e548dccf41714638a6e5862030e68bfae/docs/map-refseq.html" target="_blank">8b34f19</a>
</td>
<td>
MaevaTecher
</td>
<td>
2022-10-30
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MaevaTecher/locust-phase-transition-RNAseq/bd04bb56b25f982ed729cec095ff5ada3eb61535/docs/map-refseq.html" target="_blank">bd04bb5</a>
</td>
<td>
MaevaTecher
</td>
<td>
2022-10-30
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MaevaTecher/locust-phase-transition-RNAseq/61f4b75dd51dd84ad06beaa4a5701c0e58c0c22a/docs/map-refseq.html" target="_blank">61f4b75</a>
</td>
<td>
MaevaTecher
</td>
<td>
2022-10-30
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MaevaTecher/locust-phase-transition-RNAseq/blob/f65c630ff5f9a5de44bcfdfb29dfbf01e606946a/analysis/map-refseq.Rmd" target="_blank">f65c630</a>
</td>
<td>
MaevaTecher
</td>
<td>
2022-10-30
</td>
<td>
wflow_publish(c("analysis/_site.yml", "analysis/gene-quant.Rmd",
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="load-r-libraries-install-first-from-cran-or-bioconductor"
class="section level3">
<h3>Load R libraries (install first from CRAN or Bioconductor)</h3>
<pre class="r"><code>library(&quot;knitr&quot;)
library(&quot;rmdformats&quot;)
library(&quot;tidyverse&quot;)
library(&quot;DT&quot;)  # for making interactive search table
library(&quot;plotly&quot;) # for interactive plots
library(&quot;ggthemes&quot;) # for theme_calc
library(&quot;reshape2&quot;)

## Global options
options(max.print=&quot;10000&quot;)
knitr::opts_chunk$set(
    echo = TRUE,
    message = FALSE,
    warning = FALSE,
    cache = FALSE,
    comment = FALSE,
    prompt = FALSE,
    tidy = TRUE
)
opts_knit$set(width=75)</code></pre>
</div>
<div id="short-reads-mapping-and-gene-quantification"
class="section level1">
<h1>Short Reads mapping and Gene Quantification</h1>
<div id="star-only" class="section level2">
<h2>STAR only</h2>
<p>We used <code>STAR</code> for mapping reads to either 1) their own
species reference genome or 2) an alternate sister reference genome. The
pipeline is the same, except that the code will change index.</p>
<p>We index each genome using the following bash script:</p>
<pre><code>#!/bin/bash

##NECESSARY JOB SPECIFICATIONS
#SBATCH --job-name=STARindex        #Set the job name to &quot;JobExample4&quot;
#SBATCH --time=06:00:00         #Set the wall clock limit to 1hr and 30min
#SBATCH --ntasks=2              #Request 1 task
#SBATCH --cpus-per-task=4       #Request 1 task
#SBATCH --mem=30G              #Request 100GB per node

module purge
module load GCC/11.2.0 STAR/2.7.9a

REFDIR=&quot;/scratch/user/maeva-techer/refgenomes/locusts_complete/index_rRNA_gregaria/STAR&quot;
GENOME=&quot;/scratch/user/maeva-techer/refgenomes/locusts_complete/rRNA_gregaria.fasta&quot;
ANNOTATION=&quot;/scratch/user/maeva-techer/refgenomes/locusts_complete/rRNA_gregaria.gtf&quot;

STAR --runMode genomeGenerate --runThreadN 8 --genomeDir $REFDIR --genomeFastaFiles $GENOME --genomeSAindexNbases 10 --sjdbGTFfile $ANNOTATION --sjdbGTFfeatureExon exon --sjdbOverhang 149
</code></pre>
<p>The parameters were chosen for the following reasons:<br />
<code>--runMode genomeGenerate</code> indicates we are in the mode to
build genome index<br />
<code>--genomeSAindexNbases 22</code> should be
min(18,log2[max(GenomeLength/NumberOfReferences,ReadLength)]) according
to the manual. We have a genome of ~9000000000 and with around 1475
contigs so our value is around 22.<br />
<code>--sjdbGTFfile $ANNOTATION --sjdbGTFfeatureExon exon</code> we use
these parameters to indicates that we want the annotation file to be
already accounted for.<br />
<code>--sjdbOverhang 149</code> this should be ideally (mate_length -1),
and our reads are PE150.</p>
<pre><code>########################################
# Snakefile rule
########################################

#Ahead of the alignment I will build independently the index for STAR, HiSat2 and Segemehl

rule STAR_align:
    input:
        index = REFdir + &quot;/locusts_complete/index_{genome}/STAR&quot;,
        annotation = REFdir + &quot;/locusts_complete/{genome}.gtf&quot;,
        read1 = OUTdir + &quot;/trimming/{locust}_trim1P_1.fastq.gz&quot;,
        read2 = OUTdir + &quot;/trimming/{locust}_trim2P_2.fastq.gz&quot;
    params:
        prefix = OUTdir + &quot;/alignment/STAR/{genome}/{locust}_&quot;
    output:
        OUTdir + &quot;/alignment/STAR/{genome}/{locust}_Aligned.sortedByCoord.out.bam&quot;
    shell:
        &quot;&quot;&quot;
        module load GCC/11.2.0 STAR/2.7.9a
        STAR --runThreadN 8 --genomeDir {input.index} --readFilesIn {input.read1} {input.read2} --outFileNamePrefix {params.prefix} --readFilesCommand zcat --genomeLoad NoSharedMemory --twopassMode Basic --sjdbGTFfile {input.annotation} --sjdbScore 2 --sjdbOverhang 149 --outFilterMultimapNmax 20 --alignSJoverhangMin 8 --alignSJDBoverhangMin 1 --outFilterMatchNmin 50 --outFilterMismatchNmax 999 --outFilterMismatchNoverReadLmax 0.04 --alignIntronMin 20 --alignIntronMax 1000000 --alignMatesGapMax 1000000 --outSAMunmapped Within --outFilterType BySJout --outSAMattributes NH HI AS NM MD --outSAMtype BAM SortedByCoordinate --quantMode TranscriptomeSAM GeneCounts --quantTranscriptomeBan IndelSoftclipSingleend --limitBAMsortRAM 32000000000 --seedSearchStartLmax 30 --outFilterScoreMinOverLread 0 --outFilterMatchNminOverLread 0
        &quot;&quot;&quot;

        
########################################
# Parameters in the cluster.json file
########################################

    &quot;STAR_align&quot;:
    {
        &quot;cpus-per-task&quot; : 10,
        &quot;partition&quot; : &quot;medium&quot;,
        &quot;ntasks&quot;: 1,
        &quot;mem&quot; : &quot;100G&quot;,
        &quot;time&quot;: &quot;0-08:00:00&quot;
    }</code></pre>
<p>The parameters were chosen as follow:<br />
<code>--runThreadN 8</code> indicates that we run the mapping process
using 8 threads.<br />
<code>--genomeDir {input.index}</code> indicates where the genome index
is located.<br />
<code>--readFilesIn {input.read1} {input.read2}</code> path to the PE
reads.<br />
<code>--outFileNamePrefix {params.prefix}</code> is the prefix of our
output file.<br />
<code>--readFilesCommand zcat</code> signifies that we our reads are
compressed and need to be read as fastq.gz files.<br />
<code>--genomeLoad NoSharedMemory</code><br />
<code>--twopassMode Basic</code> indicates that we wish to use a
two-passes mapping mode that will first extract the junctions and insert
them into the genome index and re-map everything during a 2nd pass. The
option basic allows us to perform this on multiple files in
parallel.<br />
<code>--sjdbGTFfile {input.annotation}</code> indicates the path of our
annotation file.<br />
<code>--sjdbScore 2</code><br />
<code>--sjdbOverhang 149</code> is the same parameter used for building
our index.<br />
<code>--outFilterMultimapNmax 20</code><br />
<code>--alignSJoverhangMin 8</code><br />
<code>--alignSJDBoverhangMin 1</code><br />
<code>--outFilterMatchNmin 50</code><br />
<code>--outFilterMismatchNmax 999</code><br />
<code>--outFilterMismatchNoverReadLmax 0.04</code><br />
<code>--alignIntronMin 20</code><br />
<code>--alignIntronMax 1000000</code><br />
<code>--alignMatesGapMax 1000000</code><br />
<code>--outSAMunmapped Within</code><br />
<code>--outFilterType BySJout</code><br />
<code>--outSAMattributes NH HI AS NM MD</code> indicates that the
temporary .sam alignment file should contain headers.
<code>--outSAMtype BAM SortedByCoordinate</code> indicates that the
output should be in a .bam format and sorted by coordines.<br />
<code>--quantMode TranscriptomeSAM GeneCounts</code> indicates that we
wish to have two outputs, one with the Read Count for each gene and one
with the gene aligned to the transcriptome only.<br />
<code>--quantTranscriptomeBan IndelSoftclipSingleend</code><br />
<code>--limitBAMsortRAM 32000000000</code><br />
<code>--seedSearchStartLmax 30</code><br />
<code>--outFilterScoreMinOverLread 0</code><br />
<code>--outFilterMatchNminOverLread 0</code></p>
<p>After mapping, we obtained alignment statistics from the
<code>*_Log.final.out</code> file and filled out the metadata table with
it.</p>
<pre><code>grep &#39;Number of input reads&#39; *_Log.final.out
grep &#39;Average input read length&#39; *_Log.final.out
grep &#39;Uniquely mapped reads number&#39; *_Log.final.out
grep &#39;Number of reads mapped to multiple loci&#39; *_Log.final.out
grep &#39;Number of reads mapped to too many loci&#39; *_Log.final.out
grep &#39;Number of reads unmapped: too many mismatches&#39; *Log.final.out
grep &#39;Number of reads unmapped: too short&#39; *Log.final.out
grep &#39;Number of reads unmapped: other&#39; *Log.final.out</code></pre>
</div>
<div id="star-only-1" class="section level2">
<h2>STAR only</h2>
</div>
<div id="quantification-using-genecount" class="section level2">
<h2>Quantification using <code>GeneCount</code></h2>
<p>We can note that the option <code>--quantMode GeneCounts</code> from
STAR produces the same output as the htseq-count tool if we used the
<code>–-sjdbGTFfile</code> option.</p>
<p>In the output file <code>{locust}_ReadsPerGene.out.tab</code> we can
obtain the read counts per gene depending if our data is
<strong>unstranded</strong> (column 2) or <strong>stranded</strong>
(columns 3 and 4).</p>
<p>column 1: gene ID<br />
column 2: counts for unstranded RNA-seq.<br />
column 3: counts for the 1st read strand aligned with RNA<br />
column 4: counts for the 2nd read strand aligned with RNA (the most
common protocol nowadays)</p>
<p>For our pilot <em>S. gregaria</em> project, we know we used Illumina
stranded kit but to check we can with the following code:</p>
<pre><code>grep -v &quot;N_&quot; {locust}_ReadsPerGene.out.tab | awk &#39;{unst+=$2;forw+=$3;rev+=$4}END{print unst,forw,rev}&#39;

#or as a loop
for i in *_ReadsPerGene.out.tab; do echo $i; grep -v &quot;N_&quot; $i | awk &#39;{unst+=$2;forw+=$3;rev+=$4}END{print unst,forw,rev}&#39;; done</code></pre>
<p>In a stranded library preparation protocol, there should be a strong
imbalance between number of reads mapped to known genes in forward
versus reverse strands. This is what we observe for example on <em>S.
cancellata</em> libraries here.</p>
<figure class="center">
<img src="assets/ReadPerGene.png" alt="module_spider" width="800px"/>
</figure>
<p><strong>PREFERRED OPTION:</strong> We need to extract in our case the
1st and 4th columns for each file.</p>
<pre><code>########################################
# Snakefile rule
########################################

#either ran the following rule
rule reads_count:
        input:
                readtable = OUTdir + &quot;/alignment/STAR2/{locust}_ReadsPerGene.out.tab&quot;,
        output:
                OUTdir + &quot;/DESeq2/counts_4thcol/{locust}_counts.txt&quot;
        shell:
                &quot;&quot;&quot;
                cut -f1,4 {input.readtable} | grep -v &quot;_&quot; &gt; {output}  
                &quot;&quot;&quot;

#or simply this loop for less core usage
# for i in $SCRATCH/locust_phase/data/alignment/STAR/*ReadsPerGene.out.tab; do echo $i; cut -f1,4 $i | grep -v &quot;_&quot; &gt; $SCRATCH/locust_phase/data/DESeq2/counts_4thcol/`basename $i ReadsPerGene.out.tab`counts.txt; done</code></pre>
<p><strong>ALTERNATIVE OPTION:</strong> We can also build a single
matrix of expression with all individuals targeted. Below is the example
for <em>S. piceifrons</em>:</p>
<pre><code>paste SPICE_*_ReadsPerGene.out.tab | grep -v &quot;_&quot; | awk &#39;{printf &quot;%s\t&quot;, $1}{for (i=4;i&lt;=NF;i+=4) printf &quot;%s\t&quot;, $i; printf &quot;\n&quot; }&#39; &gt; tmp
sed -e &quot;1igene_name\t$(ls SPICE_*ReadsPerGene.out.tab | tr &#39;\n&#39; &#39;\t&#39; | sed &#39;s/_ReadsPerGene.out.tab//g&#39;)&quot; tmp &gt; raw_counts_piceifrons_matrix.txt</code></pre>
</div>
<div id="star-and-rsem" class="section level2">
<h2>STAR and RSEM</h2>
<p>[NEED TO DO]</p>
</div>
<div id="star-and-salmon" class="section level2">
<h2>STAR and Salmon</h2>
<p>[NEED TO DO]</p>
</div>
<div id="hisat2" class="section level2">
<h2>HISAT2</h2>
<p>[NEED TO DO]</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>FALSE R version 4.2.1 (2022-06-23)
FALSE Platform: x86_64-apple-darwin17.0 (64-bit)
FALSE Running under: macOS Big Sur ... 10.16
FALSE 
FALSE Matrix products: default
FALSE BLAS:   /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRblas.0.dylib
FALSE LAPACK: /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRlapack.dylib
FALSE 
FALSE locale:
FALSE [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
FALSE 
FALSE attached base packages:
FALSE [1] stats     graphics  grDevices utils     datasets  methods   base     
FALSE 
FALSE other attached packages:
FALSE  [1] reshape2_1.4.4   ggthemes_4.2.4   plotly_4.10.0    DT_0.26         
FALSE  [5] forcats_0.5.2    stringr_1.4.1    dplyr_1.0.10     purrr_0.3.5     
FALSE  [9] readr_2.1.3      tidyr_1.2.1      tibble_3.1.8     ggplot2_3.3.6   
FALSE [13] tidyverse_1.3.2  rmdformats_1.0.4 knitr_1.40       workflowr_1.7.0 
FALSE 
FALSE loaded via a namespace (and not attached):
FALSE  [1] httr_1.4.4          sass_0.4.2          viridisLite_0.4.1  
FALSE  [4] jsonlite_1.8.3      modelr_0.1.9        bslib_0.4.0        
FALSE  [7] assertthat_0.2.1    getPass_0.2-2       googlesheets4_1.0.1
FALSE [10] cellranger_1.1.0    yaml_2.3.6          pillar_1.8.1       
FALSE [13] backports_1.4.1     glue_1.6.2          digest_0.6.30      
FALSE [16] promises_1.2.0.1    rvest_1.0.3         colorspace_2.0-3   
FALSE [19] plyr_1.8.7          htmltools_0.5.3     httpuv_1.6.6       
FALSE [22] pkgconfig_2.0.3     broom_1.0.1         haven_2.5.1        
FALSE [25] bookdown_0.29       scales_1.2.1        processx_3.7.0     
FALSE [28] whisker_0.4         later_1.3.0         tzdb_0.3.0         
FALSE [31] git2r_0.30.1        googledrive_2.0.0   generics_0.1.3     
FALSE [34] ellipsis_0.3.2      cachem_1.0.6        withr_2.5.0        
FALSE [37] lazyeval_0.2.2      cli_3.4.1           magrittr_2.0.3     
FALSE [40] crayon_1.5.2        readxl_1.4.1        evaluate_0.17      
FALSE [43] ps_1.7.1            fs_1.5.2            fansi_1.0.3        
FALSE [46] xml2_1.3.3          tools_4.2.1         data.table_1.14.4  
FALSE [49] hms_1.1.2           formatR_1.12        gargle_1.2.1       
FALSE [52] lifecycle_1.0.3     munsell_0.5.0       reprex_2.0.2       
FALSE [55] callr_3.7.2         compiler_4.2.1      jquerylib_0.1.4    
FALSE [58] rlang_1.0.6         grid_4.2.1          rstudioapi_0.14    
FALSE [61] htmlwidgets_1.5.4   rmarkdown_2.17      gtable_0.3.1       
FALSE [64] DBI_1.1.3           R6_2.5.1            lubridate_1.8.0    
FALSE [67] fastmap_1.1.0       utf8_1.2.2          rprojroot_2.0.3    
FALSE [70] stringi_1.7.8       Rcpp_1.0.9          vctrs_0.5.0        
FALSE [73] dbplyr_2.2.1        tidyselect_1.2.0    xfun_0.34</code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
