<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Maeva Techer" />

<meta name="date" content="2022-10-31" />

<title>Preparations needed for launching the analysis</title>

<script src="site_libs/header-attrs-2.17/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Locust Phase Transition RNA-seq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Bioinformatics code
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-header">Sequence data pre-processing</li>
    <li>
      <a href="workenvs-prep.html">Preparation for the analysis</a>
    </li>
    <li>
      <a href="seqdata-prep.html">Download/obtain sequencing data</a>
    </li>
    <li>
      <a href="seqdata-qc.html">Data quality control</a>
    </li>
    <li class="dropdown-header">Genome alignment and gene quantification</li>
    <li>
      <a href="map-refseq.html">Mapping reads to Schistocerca RefSeq</a>
    </li>
    <li class="dropdown-header">Data post-processing and stats</li>
    <li>
      <a href="bam-stats.html">Indexing and stats on bam files</a>
    </li>
    <li>
      <a href="deg-workflow.html">DEG workflow</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Analysis &amp; Results
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-header">Mapping success</li>
    <li>
      <a href="own-refseq.html">Using the right RefSeq genome</a>
    </li>
    <li>
      <a href="alternate-refseq.html">Using alternate RefSeq genome</a>
    </li>
    <li class="dropdown-header">Differential Gene Expression</li>
    <li>
      <a href="deg-tissue.html">DEG per tissue</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Species
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-header">Schistocerca Locusts</li>
    <li>
      <a href="deg-gregaria.html">Desert Locust</a>
    </li>
    <li>
      <a href="deg-piceifrons.html">Central American Locust</a>
    </li>
    <li>
      <a href="deg-cancellata.html">South American Locust</a>
    </li>
    <li class="dropdown-header">Schistocerca Grasshoppers</li>
    <li>
      <a href="deg-americana.html">American Bird Grasshopper</a>
    </li>
    <li>
      <a href="deg-cubense.html">Cubense Bird Grasshopper</a>
    </li>
    <li>
      <a href="deg-nitens.html">Vagrant Grasshopper</a>
    </li>
  </ul>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/MaevaTecher/locust-phase-transition-RNAseq">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Preparations needed for launching the
analysis</h1>
<h4 class="author">Maeva Techer</h4>
<h4 class="date">2022-10-31</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-10-31
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong>
<code>locust-phase-transition-RNAseq/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20221025code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20221025)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20221025code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20221025)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomMaevaTecherlocustphasetransitionRNAseqtree1e2e9aedf80709f001a6fabf60842a2e1bf6c0eftargetblank1e2e9aea">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/MaevaTecher/locust-phase-transition-RNAseq/tree/1e2e9aedf80709f001a6fabf60842a2e1bf6c0ef" target="_blank">1e2e9ae</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomMaevaTecherlocustphasetransitionRNAseqtree1e2e9aedf80709f001a6fabf60842a2e1bf6c0eftargetblank1e2e9aea"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/MaevaTecher/locust-phase-transition-RNAseq/tree/1e2e9aedf80709f001a6fabf60842a2e1bf6c0ef" target="_blank">1e2e9ae</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    analysis/.DS_Store
    Ignored:    data/.DS_Store
    Ignored:    data/americana/.DS_Store
    Ignored:    data/americana/STAR_counts_4thcol/.DS_Store
    Ignored:    data/cancellata/.DS_Store
    Ignored:    data/cancellata/STAR_counts_4thcol/.DS_Store
    Ignored:    data/cubense/.DS_Store
    Ignored:    data/cubense/STAR_counts_4thcol/.DS_Store
    Ignored:    data/gregaria/.DS_Store
    Ignored:    data/gregaria/STAR_counts_4thcol/.DS_Store
    Ignored:    data/gregaria/list/.DS_Store
    Ignored:    data/metadata/.DS_Store
    Ignored:    data/nitens/.DS_Store
    Ignored:    data/nitens/STAR_counts_4thcol/.DS_Store
    Ignored:    data/piceifrons/.DS_Store
    Ignored:    data/piceifrons/STAR_counts_4thcol/.DS_Store
    Ignored:    data/piceifrons/list/.DS_Store

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/workenvs-prep.Rmd</code>) and
HTML (<code>docs/workenvs-prep.html</code>) files. If you’ve configured
a remote Git repository (see <code>?wflow_git_remote</code>), click on
the hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MaevaTecher/locust-phase-transition-RNAseq/ea70380071caa0d86932413a6c0120d766a79768/docs/workenvs-prep.html" target="_blank">ea70380</a>
</td>
<td>
MaevaTecher
</td>
<td>
2022-10-31
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MaevaTecher/locust-phase-transition-RNAseq/blob/de64b6bfd0f4a47e2f2cae212c8eb07a05a30c10/analysis/workenvs-prep.Rmd" target="_blank">de64b6b</a>
</td>
<td>
MaevaTecher
</td>
<td>
2022-10-31
</td>
<td>
wflow_publish(c("analysis/_site.yml", "analysis/map-refseq.Rmd",
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MaevaTecher/locust-phase-transition-RNAseq/8b34f19e548dccf41714638a6e5862030e68bfae/docs/workenvs-prep.html" target="_blank">8b34f19</a>
</td>
<td>
MaevaTecher
</td>
<td>
2022-10-30
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MaevaTecher/locust-phase-transition-RNAseq/blob/0c89c5d5e9855bdc36345b64cd624ce396f278be/analysis/workenvs-prep.Rmd" target="_blank">0c89c5d</a>
</td>
<td>
MaevaTecher
</td>
<td>
2022-10-30
</td>
<td>
wflow_publish(c("analysis/workenvs-prep.Rmd"), republish = TRUE)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MaevaTecher/locust-phase-transition-RNAseq/bd04bb56b25f982ed729cec095ff5ada3eb61535/docs/workenvs-prep.html" target="_blank">bd04bb5</a>
</td>
<td>
MaevaTecher
</td>
<td>
2022-10-30
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MaevaTecher/locust-phase-transition-RNAseq/61f4b75dd51dd84ad06beaa4a5701c0e58c0c22a/docs/workenvs-prep.html" target="_blank">61f4b75</a>
</td>
<td>
MaevaTecher
</td>
<td>
2022-10-30
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MaevaTecher/locust-phase-transition-RNAseq/blob/f65c630ff5f9a5de44bcfdfb29dfbf01e606946a/analysis/workenvs-prep.Rmd" target="_blank">f65c630</a>
</td>
<td>
MaevaTecher
</td>
<td>
2022-10-30
</td>
<td>
wflow_publish(c("analysis/_site.yml", "analysis/gene-quant.Rmd",
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="work-environment" class="section level1">
<h1>Work environment</h1>
<div id="environment" class="section level2">
<h2>Environment</h2>
<ul>
<li>Grace Cluster at Texas A&amp;M University, College Station, Texas,
USA</li>
<li>Local Mac: macOS Monterey Version 12.6<br />
</li>
<li>R version 4.2.1</li>
</ul>
</div>
<div id="directories" class="section level2">
<h2>Directories</h2>
<table>
<colgroup>
<col width="51%" />
<col width="48%" />
</colgroup>
<thead>
<tr class="header">
<th>Path</th>
<th>Purpose</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><code>/scratch/user/XXXX/locust-phase</code></td>
<td>Working repository</td>
</tr>
<tr class="even">
<td><code>/scratch/user/XXXX/locust-phase/workflow</code></td>
<td>Location of the Snakemake pipeline</td>
</tr>
<tr class="odd">
<td><code>/scratch/user/XXXX/refgenomes/locust-complete</code></td>
<td>Reference genome, annotation and index</td>
</tr>
<tr class="even">
<td><code>/scratch/user/XXXX/locust-phase/data/reads</code></td>
<td>Raw FASTQ.GZ files</td>
</tr>
<tr class="odd">
<td><code>/scratch/user/XXXX/locust-phase/data/kaiju</code></td>
<td>Kaiju database and output</td>
</tr>
<tr class="even">
<td><code>/scratch/user/XXXX/locust-phase/data/trimming</code></td>
<td>Trimmed FASTQ files</td>
</tr>
<tr class="odd">
<td><code>/scratch/user/XXXX/locust-phase/data/alignment/STAR</code></td>
<td>Output from STAR</td>
</tr>
<tr class="even">
<td><code>/scratch/user/XXXX/locust-phase/data/alignment/STAR_SALMON</code></td>
<td>Output from STAR-Salmon</td>
</tr>
<tr class="odd">
<td><code>/scratch/user/XXXX/locust-phase/data/alignment/STAR_RSEM</code></td>
<td>Output from STAR-Salmon</td>
</tr>
<tr class="even">
<td><code>/scratch/user/XXXX/locust-phase/data/analysis</code></td>
<td>R analysis (DGE)</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="snakemake-pipeline" class="section level1">
<h1>Snakemake pipeline</h1>
<p>We use a Snakemake pipeline for each species. Therefore, it is
essential to verify that each software is installed 1) locally or via a
conda environment or 2) as a module or cloned via Github on the Grace
cluster at Texas A&amp;M University is essential.</p>
<p><img src="assets/Workflow_bioinformatics.png" /></p>
<p>We built our Snakemake pipeline by launching small individual jobs to
tailor each cluster parameter for the best memory and time
efficiency.</p>
<p>To ease the indexing of our file and folder, we generate some shared
parameters which will be helpful in the future: 1) reference genome
directory path <code>REFdir</code>, 2) output directory path
<code>OUTdir</code> and 3) a list <code>LOCUSTS</code> containing sample
base name referred as <code>locust</code>.</p>
<pre><code>### SET DIRECTORY PATHS FOR REFERENCE AND OUTPUT DATA

REFdir = &quot;/scratch/user/maeva-techer/refgenomes&quot;
OUTdir = &quot;/scratch/user/maeva-techer/locust-rna/data&quot;

### SAMPLES LIST AND OTHER PARAMETERS

LOCUSTS, = glob_wildcards(OUTdir + &quot;/reads/{locust}_1.fastq.gz&quot;)
print(LOCUSTS)</code></pre>
<div id="softwares-and-database" class="section level2">
<h2>Softwares and database</h2>
<p>Links to each software to add in the future</p>
<div id="use-a-conda-environment" class="section level3">
<h3>Use a conda environment</h3>
<blockquote>
<p>add what to install on the Macpro tower e.g., we install a <a
href="https://docs.conda.io/en/latest/miniconda.html#">conda
environment</a> called <strong>rna-seq</strong></p>
</blockquote>
</div>
<div id="use-modules-on-grace-cluster" class="section level3">
<h3>Use modules on Grace cluster</h3>
<p>On Grace, each module may requires some dependencies, which is why we
need to ensure they will be loaded together. For this we use the
function <code>module spider</code> [targeted software] w/o the
version.</p>
<p><img src="assets/module_spider.png" /> Today we will need the
following software:</p>
<ul>
<li>to launch the Snakemake pipeline</li>
</ul>
<pre><code>module load GCC/11.2.0 OpenMPI/4.1.1 snakemake/6.10.0 Biopython/1.79</code></pre>
<ul>
<li>to perform adapter trimming and QC</li>
</ul>
<pre><code>module load Trimmomatic/0.39-Java-11
module load FastQC/0.11.9-Java-11</code></pre>
<ul>
<li>to perform short read mapping</li>
</ul>
<pre><code>module load GCC/11.2.0 STAR/2.7.9a</code></pre>
</div>
</div>
<div id="r-packages-neccessary" class="section level2">
<h2>R packages neccessary</h2>
<div id="list-of-r-packages-used" class="section level3">
<h3>List of R packages used</h3>
<p>CRAN<br />
data.table: <a href="https://cran.r-project.org/package=data.table"
class="uri">https://cran.r-project.org/package=data.table</a><br />
dplyr: <a href="https://cran.r-project.org/package=dplyr"
class="uri">https://cran.r-project.org/package=dplyr</a><br />
reshape2: <a href="https://cran.r-project.org/package=reshape2"
class="uri">https://cran.r-project.org/package=reshape2</a><br />
ggplot2: <a href="https://cran.r-project.org/package=ggplot2"
class="uri">https://cran.r-project.org/package=ggplot2</a><br />
ggrepel: <a href="https://cran.r-project.org/package=ggrepel"
class="uri">https://cran.r-project.org/package=ggrepel</a></p>
<p>Bioconductor<br />
DESeq2: <a href="https://bioconductor.org/packages/DESeq2/"
class="uri">https://bioconductor.org/packages/DESeq2/</a><br />
apeglm: <a href="https://bioconductor.org/packages/apeglm/"
class="uri">https://bioconductor.org/packages/apeglm/</a><br />
EnhancedVolcano: <a
href="https://bioconductor.org/packages/EnhancedVolcano/"
class="uri">https://bioconductor.org/packages/EnhancedVolcano/</a><br />
edgeR: <a href="https://www.bioconductor.org/packages/edgeR/"
class="uri">https://www.bioconductor.org/packages/edgeR/</a></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.1 (2022-06-23)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur ... 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.2/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] workflowr_1.7.0

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.9       compiler_4.2.1   pillar_1.8.1     bslib_0.4.0     
 [5] later_1.3.0      git2r_0.30.1     jquerylib_0.1.4  tools_4.2.1     
 [9] getPass_0.2-2    digest_0.6.30    jsonlite_1.8.3   evaluate_0.17   
[13] tibble_3.1.8     lifecycle_1.0.3  pkgconfig_2.0.3  rlang_1.0.6     
[17] cli_3.4.1        rstudioapi_0.14  yaml_2.3.6       xfun_0.34       
[21] fastmap_1.1.0    httr_1.4.4       stringr_1.4.1    knitr_1.40      
[25] fs_1.5.2         vctrs_0.5.0      sass_0.4.2       rprojroot_2.0.3 
[29] glue_1.6.2       R6_2.5.1         processx_3.7.0   fansi_1.0.3     
[33] rmarkdown_2.17   callr_3.7.2      magrittr_2.0.3   whisker_0.4     
[37] ps_1.7.1         promises_1.2.0.1 htmltools_0.5.3  httpuv_1.6.6    
[41] utf8_1.2.2       stringi_1.7.8    cachem_1.0.6    </code></pre>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
