---
title: "Differential Gene Expression using DEseq2"
author: "Maeva Techer"
date: "`r Sys.Date()`"
output:
  workflowr::wflow_html
---

# Load required R libraries 
```{r libraries, message=FALSE, warning=FALSE}
#(install first from CRAN or Bioconductor)
library("knitr")
library("rmdformats")
library("tidyverse")
library("DT")  # for making interactive search table
library("plotly") # for interactive plots
library("ggthemes") # for theme_calc
library("reshape2")
library("DESeq2")
library("data.table")
library("apeglm")
library("ggpubr")
library("ggplot2")
library("ggrepel")
library("EnhancedVolcano")
library("SARTools")
library("pheatmap")

## Global options
options(max.print="10000")
knitr::opts_chunk$set(
	echo = TRUE,
	message = FALSE,
	warning = FALSE,
	cache = FALSE,
	comment = FALSE,
	prompt = FALSE,
	tidy = TRUE
)
opts_knit$set(width=75)
```

Here we present the workflow example with the head data from _S. piceifrons_  
  
For the analysis of differentially expressed genes, we will follow some guidelines from an online RNA course [tutorial](https://biocorecrg.github.io/RNAseq_course_2019/differential_expression.html) that uses either `DESeq2` or `edgeR` on STAR output. We also adapted some script lines from Foquet et al. 2021 code.

DESeq2 tests for differential expression using negative binomial generalized linear models.
DESeq2 (as edgeR) is based on the hypothesis that most genes are not differentially expressed. The package takes as an input raw counts (i.e. non normalized counts): the DESeq2 model internally corrects for library size, so giving as an input normalized count would be incorrect.

# DESeq2 analysis using STAR input
## Preparing input files 
#### Raw count matrices

We generated this in the precedent section.

### Transcript-to-gene annotation file
  
Below is the example with _S. piceifrons_
```
# Download annotation and place it into the folder refgenomes
wget https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/021/461/385/GCF_021461385.2_iqSchPice1.1/GCF_021461385.2_iqSchPice1.1_genomic.gtf.gz

# first column is the transcript ID, second column is the gene ID, third column is the gene symbol
zcat GCF_021461385.2_iqSchPice1.1_genomic.gtf.gz | awk -F "\t" 'BEGIN{OFS="\t"}{if($3=="transcript"){split($9, a, "\""); print a[4],a[2],a[8]}}' > tx2gene.piceifrons.csv
```
  
### Sample sheet 
  
DESeq2 needs a sample sheet that describes the samples characteristics: SampleName, FileName (...counts.txt), and subsequently anything that can be used for **statistical design** such as RearingCondition, replicates, tissue, time points, etc. in the form. 
  
The design indicates how to model the samples: in the model we need to specify what we want to measure and what we want to **control**.  

## Differential expression analysis
### Import input count data

We start by reading the sample sheet.
```{r DESeq2 prep, message=FALSE, warning=FALSE}

###############################
#  MOSTLY FOR THE HMTL REPORT
###############################
## PARAMETERS for running the script
homeDir <- "/Users/alphamanae/Documents/GitHub/locust-phase-transition-RNAseq"
workDir <- "/Users/alphamanae/Documents/GitHub/locust-phase-transition-RNAseq/data/piceifrons"  		# Working directory
projectName <- "SPICE_HEAD"             			# name of the project
author <- "Maeva TECHER"                    	# author of the statistical analysis/report

## Create all the needed directories
setwd(workDir)
Dirname <- paste("DEseq2_", projectName, sep = "")
dir.create(Dirname)
setwd(Dirname)
workDir_DEseq2 <- getwd()
# path to the directory containing raw counts files
rawDir <- "/Users/alphamanae/Documents/GitHub/locust-phase-transition-RNAseq/data/piceifrons/STAR_counts_4thcol"              

## PARAMETERS for running DEseq2
tresh_logfold <- 1					          # Treshold for log2(foldchange) in final DE-files
tresh_padj <- 0.05					          # Treshold for adjusted p-valued in final DE-files
alpha_DEseq2 <- 0.05                  # threshold of statistical significance
pAdjustMethod_DEseq2 <- "BH"          # p-value adjustment method: "BH" (default) or "BY"
featuresToRemove <- c(NULL)           # names of the features to be removed, NULL if none or if using Idxstats
varInt <- "RearingCondition"          # factor of interest
condRef <- "Isolated"                 # reference biological condition
batch <- NULL                         # blocking factor: NULL (default) or "batch" for example  
fitType <- "parametric"               # mean-variance relationship: "parametric" (default) or "local"
cooksCutoff <- TRUE                   # TRUE/FALSE to perform the outliers detection (default is TRUE)
independentFiltering <- TRUE          # TRUE/FALSE to perform independent filtering (default is TRUE)
typeTrans <- "VST"                   	# transformation for PCA/clustering: "VST" or "rlog"
locfunc <- "median"

# Path and name of targetfile containing conditions and file names
targetFile <- "/Users/alphamanae/Documents/GitHub/locust-phase-transition-RNAseq/data/piceifrons/list/HeadSPICE.txt"           
colors <- c("#B31B21", "#1465AC")

# checking parameters
setwd(workDir_DEseq2)
checkParameters.DESeq2(projectName=projectName,author=author,targetFile=targetFile,
                       rawDir=rawDir,featuresToRemove=featuresToRemove,varInt=varInt,
                       condRef=condRef,batch=batch,fitType=fitType,cooksCutoff=cooksCutoff,
                       independentFiltering=independentFiltering,alpha=alpha,pAdjustMethod=pAdjustMethod_DEseq2,
                       typeTrans=typeTrans,locfunc=locfunc,colors=colors)



###############################
###############################

setwd(homeDir)
sampletable <- fread("data/piceifrons/list/HeadSPICE.txt")
## add the sample names as row names (it is needed for some of the DESeq functions)
rownames(sampletable) <- sampletable$SampleName

## Make sure discriminant variables are factor
sampletable$RearingCondition <- as.factor(sampletable$RearingCondition)
sampletable$Tissue <- as.factor(sampletable$Tissue)
dim(sampletable)

```

Then we obtain the output from `STAR GeneCount` and import here individually using the sampletable as a reference to fetch them. We also filter the lowly expressed genes to avoid noisy data.

```{r DESeq2 import, message=FALSE, warning=FALSE}

## Import count files
satoshi <- DESeqDataSetFromHTSeqCount(sampleTable = sampletable,
                        directory = "data/piceifrons/STAR_counts_4thcol",
                        design = ~ RearingCondition)
satoshi

# keep genes for which sums of raw counts across experimental samples is > 5
satoshi <- satoshi[rowSums(counts(satoshi)) > 5, ]
nrow(satoshi)

#set a standard to be compared to (hatchling) ONLY IF WE HAVE A CONTROL
#satoshi$Tissue <- relevel(satoshi$Tissue, ref = "Whole_body")
```

### Run the model Fit DESeq2

Run DESeq2 analysis using `DESeq`, which performs (1) estimation of size factors, (2) estimation of dispersion, then (3) Negative Binomial GLM fitting and Wald statistics. The results tables (log2 fold changes and p-values) can be generated using the results function ([copied from online chapter](https://ycl6.gitbook.io/guide-to-rna-seq-analysis/differential-expression-analysis/differential-gene-expression/dge-analysis-with-star-input))

```{r DESeq2 model, message=FALSE, warning=FALSE}

# Fit the statistical model
shigeru <- DESeq(satoshi)
cbind(resultsNames(shigeru))

# Here we plot the adjusted p-value means corrected for multiple testing (FDR padj)
res_shigeru <- results(shigeru)

# We only keep genes with an adjusted p-value cutoff > 0.05 by changing the default significance cut-off
sum(res_shigeru$padj < tresh_padj, na.rm = TRUE)

brock <- results(shigeru, name = "RearingCondition_Isolated_vs_Crowded", alpha = alpha_DEseq2)
summary(brock)

# Details of what is each column meaning in our final result 
mcols(brock)$description
head(brock)

```

For this data set after FDR filtering of 0.05, we have 254 genes up-regulates and 404 genes down-regulated in crowded versus solitary individuals.

## Visualizing and exploring the results
### PCA of the samples

We transformed the data for visualization by comparing both recommended **rlog** (Regularized log) or **vst** (Variance Stabilizing Transformation) transformations. Both options produce log2 scale data which has been normalized by the DESeq2 method with respect to library size.

```{r Transform data, message=FALSE, warning=FALSE}
# Try with the vst transformation
shigeru_vst <- vst(shigeru)
shigeru_rlog <- rlog(shigeru)
```

Plot the PCA rlog

```{r DESeq2 PCA rlog, message=FALSE, warning=FALSE}

# Create the pca on the defined groups
pcaData <- plotPCA(object = shigeru_rlog, intgroup = c("RearingCondition", "Tissue"),returnData=TRUE)

# Store the information for each axis variance in %
percentVar <- round(100 * attr(pcaData, "percentVar"))

# Make sure that the discriminant variable are in factor for using shape and color later on
pcaData$RearingCondition<-factor(pcaData$RearingCondition,levels=c("Crowded","Isolated"), labels=c("crowded piceifrons","isolated piceifrons"))
levels(pcaData$RearingCondition)
#pcaData$Tissue<-factor(pcaData1$Tissue,levels=c("Whole_body","Optical_lobes"), labels=c("Hatchling", "OLB"))
#levels(pcaData$Tissue)


ggplot(pcaData, aes(PC1, PC2, color= RearingCondition)) +
  geom_point(size=4)+xlab(paste0("PC1: ",percentVar[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar[2],"% variance")) + 
  scale_color_manual(values = c("blue", "red")) +
  geom_text_repel(aes(label = name), nudge_x = -1, nudge_y = 0.2, size = 3) +
  coord_fixed()+
  theme_bw()+
  theme(legend.title = element_blank())+ 
  theme(legend.text = element_text(face="bold", size=15))+
  theme(axis.text = element_text(size=15))+
  theme(axis.title = element_text(size=16))+ 
  ggtitle("PCA on S. piceifrons head tissues", subtitle = "rlog transformation") + 
  xlab(paste0("PC1: ",percentVar[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar[2],"% variance"))
```

Plot the PCA vsd

```{r DESeq2 PCA vsd, message=FALSE, warning=FALSE}

# Create the pca on the defined groups
pcaData <- plotPCA(object = shigeru_vst, intgroup = c("RearingCondition", "Tissue"),returnData=TRUE)

# Store the information for each axis variance in %
percentVar <- round(100 * attr(pcaData, "percentVar"))

# Make sure that the discriminant variable are in factor for using shape and color later on
pcaData$RearingCondition<-factor(pcaData$RearingCondition,levels=c("Crowded","Isolated"), labels=c("crowded piceifrons","isolated piceifrons"))
levels(pcaData$RearingCondition)
#pcaData$Tissue<-factor(pcaData1$Tissue,levels=c("Whole_body","Optical_lobes"), labels=c("Hatchling", "OLB"))
#levels(pcaData$Tissue)


ggplot(pcaData, aes(PC1, PC2, color= RearingCondition)) +
  geom_point(size=4)+xlab(paste0("PC1: ",percentVar[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar[2],"% variance")) + 
  scale_color_manual(values = c("blue", "red")) +
  geom_text_repel(aes(label = name), nudge_x = -1, nudge_y = 0.2, size = 3) +
  coord_fixed()+
  theme_bw()+
  theme(legend.title = element_blank())+ 
  theme(legend.text = element_text(face="bold", size=15))+
  theme(axis.text = element_text(size=15))+
  theme(axis.title = element_text(size=16))+ 
  ggtitle("PCA on S. piceifrons head tissues", subtitle = "vst transformation") + 
  xlab(paste0("PC1: ",percentVar[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar[2],"% variance"))
```

### Sample Matrix Distance

Using also the transformed data, we check the distance between samples and see how they correlate to each others.

Heatmap using rlog

```{r DESeq2 distance samples, message=FALSE, warning=FALSE}

# calculate between-sample distance matrix
sampleDistMatrix.rlog <- as.matrix(dist(t(assay(shigeru_rlog))))

metadata <- sampletable[,c("RearingCondition", "Tissue")]
rownames(metadata) <- sampletable$SampleName


pheatmap(sampleDistMatrix.rlog, annotation_col=metadata, main = "Head tissue heatmap, rlog transformation")

```

Heatmap using vst

```{r DESeq2 distance samples2, message=FALSE, warning=FALSE}

# calculate between-sample distance matrix
sampleDistMatrix.vst <- as.matrix(dist(t(assay(shigeru_vst))))

pheatmap(sampleDistMatrix.vst, annotation_col=metadata, main = "Head tissue heatmap, vst transformation")

```

### MA-plot

This plot allows us to show the log2 fold changes over the mean of normalized counts for all the samples. Points will be colored in red if the adjusted p-value is less than 0.05 and the log2 fold change is bigger than 1. In blue, will be the reverse for the log2 fold change.

To generate more accurate log2 foldchange estimates, DESeq2 allows (and recommends) the shrinkage of the LFC estimates toward zero when the information for a gene is low, which could include:  
  
-Low counts  
-High dispersion values  

```{r DESeq2 MA plot , message=FALSE, warning=FALSE}

# include the log2FoldChange shrinkage use to visualize gene ranking
de_shrink <- lfcShrink(dds = shigeru, coef="RearingCondition_Isolated_vs_Crowded", type="apeglm")
head(de_shrink)

# Ma plot
maplot <-ggmaplot(de_shrink, fdr = 0.05, fc = 1, size = 2, palette = c("#B31B21", "#1465AC", "darkgray"), genenames = as.vector(rownames(de_shrink$name)), top = 0,legend="top",label.select = NULL) +
coord_cartesian(xlim = c(0, 20)) + 
scale_y_continuous(limits=c(-12, 12)) + 
theme(axis.text.x = element_text(size=16),axis.text.y = element_text(size=15),axis.title.x = element_text(size=17),axis.title.y = element_text(size=17),axis.line = element_line(size = 1, colour="gray20"),axis.ticks = element_line(size = 1, colour="gray20"))+
guides(color = guide_legend(override.aes = list(size = c(3,3,3))))+
theme(legend.position = c(0.70, 0.12),legend.text=element_text(size=14,face="bold"),legend.background = element_rect(fill="transparent"))+
theme(plot.title = element_text(size=18, colour="gray30", face="bold",hjust=0.06, vjust=-5))+
labs(title="MA-plot for the shrunken log2 fold changes in the head")
maplot
```

### Volcano Plot

The `EnhancedVolcano` helps visualise the resulst of differential expression analysis.

```{r DESeq2 Volcano plot , message=FALSE, warning=FALSE}
keyvals <-ifelse( 
  res_shigeru$log2FoldChange >= 1 & res_shigeru$padj <= 0.05, '#B31B21', 
  ifelse(res_shigeru$log2FoldChange <= -1 & res_shigeru$padj <= 0.05, '#1465AC', 'darkgray')) 

keyvals[is.na(keyvals)] <-'darkgray' 
names(keyvals)[keyvals == "#B31B21"] <-'Upregulated' 
names(keyvals)[keyvals == "#1465AC"] <-'Downregulated' 
names(keyvals)[keyvals == 'darkgray'] <-'NS'

EnhancedVolcano(res_shigeru,
                lab = rownames(res_shigeru),
                x = 'log2FoldChange',
                y = 'padj',
                pCutoff = 0.05,
                FCcutoff = 1,
                pointSize = 3,
                labSize = 4,
                colAlpha = 4/5,
                colCustom = keyvals,
                drawConnectors = TRUE)
```

### Normalized Matrix distance

We then normalize the result by extracting only significant genes with a fold change of 1.


```{r DESeq2 norm heatmap, message=FALSE, warning=FALSE}

resorted_deresults <- res_shigeru[order(res_shigeru$padj),]

## Select only the genes that have a padj > 0.05 and with minimum log2FoldChange of 1
sig <- resorted_deresults[!is.na(resorted_deresults$padj) &
                            resorted_deresults$padj < tresh_padj &
                            abs(resorted_deresults$log2FoldChange) >= tresh_logfold,]
selected <- rownames(sig);selected

## Norm transform the data from DEseq2 run
ntd <- normTransform(satoshi)

## Plot the relation among samples considering only the significant genes
pheatmap(assay(ntd)[selected,], 
         cluster_rows = TRUE, 
         show_rownames = TRUE, 
         cluster_cols = TRUE, 
         cutree_rows = 4,
         cutree_cols = 3,
         labels_col= colData(satoshi)$SampleName)
```


## Create a hmtl report with DEseq2

```{r DESeq2 html report version , message=FALSE, warning=FALSE}

## import the sample sheet that indicates Rearing Conditions and Tissue origins
setwd(workDir_DEseq2)
# loading target file
target <- loadTargetFile(targetFile=targetFile, varInt=varInt, condRef=condRef, batch=batch)

# loading counts
counts <- loadCountData(target=target, rawDir=rawDir, featuresToRemove=featuresToRemove)

# description plots
majSequences <- descriptionPlots(counts=counts, group=target[,varInt], col=colors)

# analysis with DESeq2
out.DESeq2 <- run.DESeq2(counts=counts, target=target, varInt=varInt, batch=batch,
                         locfunc=locfunc, fitType=fitType, pAdjustMethod=pAdjustMethod_DEseq2,
                         cooksCutoff=cooksCutoff, independentFiltering=independentFiltering, alpha=alpha_DEseq2)

# PCA + Clustering
exploreCounts(object=out.DESeq2$dds, group=target[,varInt], typeTrans=typeTrans, col=colors)

# summary of the analysis (boxplots, dispersions, diag size factors, export table, nDiffTotal, histograms, MA plot)
summaryResults <- summarizeResults.DESeq2(out.DESeq2, group=target[,varInt], col=colors,
                                          independentFiltering=independentFiltering,
                                          cooksCutoff=cooksCutoff, alpha=alpha_DEseq2)

# generating HTML report
writeReport.DESeq2(target=target, counts=counts, out.DESeq2=out.DESeq2, summaryResults=summaryResults,
                   majSequences=majSequences, workDir=workDir_DEseq2, projectName=projectName, author=author,
                   targetFile=targetFile, rawDir=rawDir, featuresToRemove=featuresToRemove, varInt=varInt,
                   condRef=condRef, batch=batch, fitType=fitType, cooksCutoff=cooksCutoff,
                   independentFiltering=independentFiltering, alpha=alpha_DEseq2, pAdjustMethod=pAdjustMethod_DEseq2,
                   typeTrans=typeTrans, locfunc=locfunc, colors=colors)

```
