---
title: "Mapping reads to Schistocerca RefSeq genome"
author: "Maeva Techer"
date: "`r Sys.Date()`"
output:
  workflowr::wflow_html
---

### Load R libraries (install first from CRAN or Bioconductor)
```{r libraries, message=FALSE, warning=FALSE}
library("knitr")
library("rmdformats")
library("tidyverse")
library("DT")  # for making interactive search table
library("plotly") # for interactive plots
library("ggthemes") # for theme_calc
library("reshape2")

## Global options
options(max.print="10000")
knitr::opts_chunk$set(
	echo = TRUE,
	message = FALSE,
	warning = FALSE,
	cache = FALSE,
	comment = FALSE,
	prompt = FALSE,
	tidy = TRUE
)
opts_knit$set(width=75)
```

# Short Reads mapping and Gene Quantification
## STAR only

We used `STAR` for mapping reads to either 1) their own species reference genome or 2) an alternate sister reference genome. The pipeline is the same, except that the code will change index.

We index each genome using the following bash script:  

```
#!/bin/bash

##NECESSARY JOB SPECIFICATIONS
#SBATCH --job-name=STARindex       	#Set the job name to "JobExample4"
#SBATCH --time=06:00:00         #Set the wall clock limit to 1hr and 30min
#SBATCH --ntasks=2              #Request 1 task
#SBATCH --cpus-per-task=4       #Request 1 task
#SBATCH --mem=30G              #Request 100GB per node

module purge
module load GCC/11.2.0 STAR/2.7.9a

REFDIR="/scratch/user/maeva-techer/refgenomes/locusts_complete/index_rRNA_gregaria/STAR"
GENOME="/scratch/user/maeva-techer/refgenomes/locusts_complete/rRNA_gregaria.fasta"
ANNOTATION="/scratch/user/maeva-techer/refgenomes/locusts_complete/rRNA_gregaria.gtf"

STAR --runMode genomeGenerate --runThreadN 8 --genomeDir $REFDIR --genomeFastaFiles $GENOME --genomeSAindexNbases 10 --sjdbGTFfile $ANNOTATION --sjdbGTFfeatureExon exon --sjdbOverhang 149

```

The parameters were chosen for the following reasons:  
`--runMode genomeGenerate` indicates we are in the mode to build genome index  
`--genomeSAindexNbases 22` should be  min(18,log2[max(GenomeLength/NumberOfReferences,ReadLength)]) according to the manual. We have a genome of ~9000000000 and with around 1475 contigs so our value is around 22.  
`--sjdbGTFfile $ANNOTATION --sjdbGTFfeatureExon exon` we use these parameters to indicates that we want the annotation file to be already accounted for.  
`--sjdbOverhang 149` this should be ideally (mate_length -1), and our reads are PE150.  

```
########################################
# Snakefile rule
########################################

#Ahead of the alignment I will build independently the index for STAR, HiSat2 and Segemehl

rule STAR_align:
	input:
		index = REFdir + "/locusts_complete/index_{genome}/STAR",
		annotation = REFdir + "/locusts_complete/{genome}.gtf",
		read1 = OUTdir + "/trimming/{locust}_trim1P_1.fastq.gz",
		read2 = OUTdir + "/trimming/{locust}_trim2P_2.fastq.gz"
	params:
		prefix = OUTdir + "/alignment/STAR/{genome}/{locust}_"
	output:
		OUTdir + "/alignment/STAR/{genome}/{locust}_Aligned.sortedByCoord.out.bam"
	shell:
		"""
		module load GCC/11.2.0 STAR/2.7.9a
		STAR --runThreadN 8 --genomeDir {input.index} --readFilesIn {input.read1} {input.read2} --outFileNamePrefix {params.prefix} --readFilesCommand zcat --genomeLoad NoSharedMemory --twopassMode Basic --sjdbGTFfile {input.annotation} --sjdbScore 2 --sjdbOverhang 149 --outFilterMultimapNmax 20 --alignSJoverhangMin 8 --alignSJDBoverhangMin 1 --outFilterMatchNmin 50 --outFilterMismatchNmax 999 --outFilterMismatchNoverReadLmax 0.04 --alignIntronMin 20 --alignIntronMax 1000000 --alignMatesGapMax 1000000 --outSAMunmapped Within --outFilterType BySJout --outSAMattributes NH HI AS NM MD --outSAMtype BAM SortedByCoordinate --quantMode TranscriptomeSAM GeneCounts --quantTranscriptomeBan IndelSoftclipSingleend --limitBAMsortRAM 32000000000 --seedSearchStartLmax 30 --outFilterScoreMinOverLread 0 --outFilterMatchNminOverLread 0
		"""

		
########################################
# Parameters in the cluster.json file
########################################

    "STAR_align":
    {
        "cpus-per-task" : 10,
        "partition" : "medium",
        "ntasks": 1,
        "mem" : "100G",
        "time": "0-08:00:00"
    }
```

The parameters were chosen as follow:  
`--runThreadN 8` indicates that we run the mapping process using 8 threads.  
`--genomeDir {input.index}` indicates where the genome index is located.  
`--readFilesIn {input.read1} {input.read2}` path to the PE reads.  
`--outFileNamePrefix {params.prefix}` is the prefix of our output file.  
`--readFilesCommand zcat` signifies that we our reads are compressed and need to be read as fastq.gz files.  
`--genomeLoad NoSharedMemory`   
`--twopassMode Basic` indicates that we wish to use a two-passes mapping mode that will first extract the junctions and insert them into the genome index and re-map everything during a 2nd pass. The option basic allows us to perform this on multiple files in parallel.  
`--sjdbGTFfile {input.annotation}` indicates the path of our annotation file.     
`--sjdbScore 2`   
`--sjdbOverhang 149` is the same parameter used for building our index.  
`--outFilterMultimapNmax 20`   
`--alignSJoverhangMin 8`   
`--alignSJDBoverhangMin 1`   
`--outFilterMatchNmin 50`   
`--outFilterMismatchNmax 999`   
`--outFilterMismatchNoverReadLmax 0.04`   
`--alignIntronMin 20`   
`--alignIntronMax 1000000`   
`--alignMatesGapMax 1000000`   
`--outSAMunmapped Within`   
`--outFilterType BySJout`   
`--outSAMattributes NH HI AS NM MD` indicates that the temporary .sam alignment file should contain headers.
`--outSAMtype BAM SortedByCoordinate` indicates that the output should be in a .bam format and sorted by coordines.     
`--quantMode TranscriptomeSAM GeneCounts` indicates that we wish to have two outputs, one with the Read Count for each gene and one with the gene aligned to the transcriptome only.    
`--quantTranscriptomeBan IndelSoftclipSingleend`   
`--limitBAMsortRAM 32000000000`   
`--seedSearchStartLmax 30`  
`--outFilterScoreMinOverLread 0`  
`--outFilterMatchNminOverLread 0`  


After mapping, we obtained alignment statistics from the `*_Log.final.out` file and filled out the metadata table with it.

```
grep 'Number of input reads' *_Log.final.out
grep 'Average input read length' *_Log.final.out
grep 'Uniquely mapped reads number' *_Log.final.out
grep 'Number of reads mapped to multiple loci' *_Log.final.out
grep 'Number of reads mapped to too many loci' *_Log.final.out
grep 'Number of reads unmapped: too many mismatches' *Log.final.out
grep 'Number of reads unmapped: too short' *Log.final.out
grep 'Number of reads unmapped: other' *Log.final.out
```

## STAR only

  
## Quantification using `GeneCount`

We can note that the option `--quantMode GeneCounts` from STAR produces the same output as the htseq-count tool if we used the `â€“-sjdbGTFfile` option.   
  
In the output file `{locust}_ReadsPerGene.out.tab` we can obtain the read counts per gene depending if our data is **unstranded** (column 2) or **stranded** (columns 3 and 4).  
  
column 1: gene ID  
column 2: counts for unstranded RNA-seq.  
column 3: counts for the 1st read strand aligned with RNA  
column 4: counts for the 2nd read strand aligned with RNA (the most common protocol nowadays)  
  

For our pilot _S. gregaria_ project, we know we used Illumina stranded kit but to check we can with the following code:  

```
grep -v "N_" {locust}_ReadsPerGene.out.tab | awk '{unst+=$2;forw+=$3;rev+=$4}END{print unst,forw,rev}'

#or as a loop
for i in *_ReadsPerGene.out.tab; do echo $i; grep -v "N_" $i | awk '{unst+=$2;forw+=$3;rev+=$4}END{print unst,forw,rev}'; done
```
  
In a stranded library preparation protocol, there should be a strong imbalance between number of reads mapped to known genes in forward versus reverse strands. This is what we observe for example on _S. cancellata_ libraries here.  
  
<figure class="center">
<img src="assets/ReadPerGene.png" alt="module_spider" width="800px"/>  
</figure>   
  
**PREFERRED OPTION:** We need to extract in our case the 1st and 4th columns for each file.

```
########################################
# Snakefile rule
########################################

#either ran the following rule
rule reads_count:
        input:
                readtable = OUTdir + "/alignment/STAR2/{locust}_ReadsPerGene.out.tab",
        output:
                OUTdir + "/DESeq2/counts_4thcol/{locust}_counts.txt"
        shell:
                """
                cut -f1,4 {input.readtable} | grep -v "_" > {output}  
                """

#or simply this loop for less core usage
# for i in $SCRATCH/locust_phase/data/alignment/STAR/*ReadsPerGene.out.tab; do echo $i; cut -f1,4 $i | grep -v "_" > $SCRATCH/locust_phase/data/DESeq2/counts_4thcol/`basename $i ReadsPerGene.out.tab`counts.txt; done
```

**ALTERNATIVE OPTION:** We can also build a single matrix of expression with all individuals targeted. Below is the example for _S. piceifrons_:

```
paste SPICE_*_ReadsPerGene.out.tab | grep -v "_" | awk '{printf "%s\t", $1}{for (i=4;i<=NF;i+=4) printf "%s\t", $i; printf "\n" }' > tmp
sed -e "1igene_name\t$(ls SPICE_*ReadsPerGene.out.tab | tr '\n' '\t' | sed 's/_ReadsPerGene.out.tab//g')" tmp > raw_counts_piceifrons_matrix.txt
```

## STAR and RSEM

[NEED TO DO]

## STAR and Salmon 

[NEED TO DO]

## HISAT2  

[NEED TO DO]
