---
title: "Mapping reads to Schistocerca RefSeq genome"
author: "Maeva Techer"
date: "`r Sys.Date()`"
output:
  workflowr::wflow_html
---

### Load R libraries (install first from CRAN or Bioconductor)
```{r libraries, message=FALSE, warning=FALSE}
library("knitr")
library("rmdformats")
library("tidyverse")
library("DT")  # for making interactive search table
library("plotly") # for interactive plots
library("ggthemes") # for theme_calc
library("reshape2")

## Global options
options(max.print="10000")
knitr::opts_chunk$set(
	echo = TRUE,
	message = FALSE,
	warning = FALSE,
	cache = FALSE,
	comment = FALSE,
	prompt = FALSE,
	tidy = TRUE
)
opts_knit$set(width=75)
```

# Short Reads mapping and Gene Quantification
## STAR only

We used `STAR` for mapping reads to either 1) their own species reference genome or 2) an alternate sister reference genome. The pipeline is the same, except that the code will change index.

```
########################################
# Snakefile rule
########################################

#Ahead of the alignment I will build independently the index for STAR, HiSat2 and Segemehl

rule STAR_align:
	input:
		index = REFdir + "/locusts_complete/index_GCF_021461395.2_iqSchAmer2.1_genomic/STAR",
		read1 = OUTdir + "/trimming/{locust}_trim1P_1.fastq.gz",
		read2 = OUTdir + "/trimming/{locust}_trim2P_2.fastq.gz"
	params:
		prefix = OUTdir + "/alignment/STAR/{locust}_"
	output:
		OUTdir + "/alignment/STAR/{locust}_Aligned.sortedByCoord.out.bam"
	shell:
		"""
		module load GCC/11.2.0 STAR/2.7.9a
		STAR --runThreadN 8 --genomeDir {input.index} --outSAMtype BAM SortedByCoordinate --quantMode GeneCounts --outFileNamePrefix {params.prefix} --readFilesCommand zcat --readFilesIn {input.read1} {input.read2} 
		"""
		
########################################
# Parameters in the cluster.json file
########################################

    "STAR_align":
    {
        "cpus-per-task" : 12,
        "partition" : "medium",
        "ntasks": 1,
        "mem" : "100G",
        "time": "0-08:00:00"
    }
```

After mapping, we obtained alignment statistics from the `*_Log.final.out` file and filled out the metadata table with it.

```
grep 'Number of input reads' *_Log.final.out
grep 'Average input read length' *_Log.final.out
grep 'Uniquely mapped reads number' *_Log.final.out
grep 'Number of reads mapped to multiple loci' *_Log.final.out
grep 'Number of reads mapped to too many loci' *_Log.final.out
grep 'Number of reads unmapped: too many mismatches' *Log.final.out
grep 'Number of reads unmapped: too short' *Log.final.out
grep 'Number of reads unmapped: other' *Log.final.out
```

## STAR only

  
## Quantification using `GeneCount`

We can note that the option `--quantMode GeneCounts` from STAR produces the same output as the htseq-count tool if we used the `â€“-sjdbGTFfile` option.   
  
In the output file `{locust}_ReadsPerGene.out.tab` we can obtain the read counts per gene depending if our data is **unstranded** (column 2) or **stranded** (columns 3 and 4).  
  
column 1: gene ID  
column 2: counts for unstranded RNA-seq.  
column 3: counts for the 1st read strand aligned with RNA  
column 4: counts for the 2nd read strand aligned with RNA (the most common protocol nowadays)  
  

For our pilot _S. gregaria_ project, we know we used Illumina stranded kit but to check we can with the following code:  

```
grep -v "N_" {locust}_ReadsPerGene.out.tab | awk '{unst+=$2;forw+=$3;rev+=$4}END{print unst,forw,rev}'

#or as a loop
for i in *_ReadsPerGene.out.tab; do echo $i; grep -v "N_" $i | awk '{unst+=$2;forw+=$3;rev+=$4}END{print unst,forw,rev}'; done
```
  
In a stranded library preparation protocol, there should be a strong imbalance between number of reads mapped to known genes in forward versus reverse strands. This is what we observe for example on _S. cancellata_ libraries here.  
  
<figure class="center">
<img src="img/ReadPerGene.png" alt="module_spider" width="800px"/>  
</figure>   
  
**PREFERRED OPTION:** We need to extract in our case the 1st and 4th columns for each file.

```
########################################
# Snakefile rule
########################################

#either ran the following rule
rule reads_count:
        input:
                readtable = OUTdir + "/alignment/STAR2/{locust}_ReadsPerGene.out.tab",
        output:
                OUTdir + "/DESeq2/counts_4thcol/{locust}_counts.txt"
        shell:
                """
                cut -f1,4 {input.readtable} | grep -v "_" > {output}  
                """

#or simply this loop for less core usage
# for i in $SCRATCH/locust_phase/data/alignment/STAR/*ReadsPerGene.out.tab; do echo $i; cut -f1,4 $i | grep -v "_" > $SCRATCH/locust_phase/data/DESeq2/counts_4thcol/`basename $i ReadsPerGene.out.tab`counts.txt; done
```

**ALTERNATIVE OPTION:** We can also build a single matrix of expression with all individuals targeted. Below is the example for _S. piceifrons_:

```
paste SPICE_*_ReadsPerGene.out.tab | grep -v "_" | awk '{printf "%s\t", $1}{for (i=4;i<=NF;i+=4) printf "%s\t", $i; printf "\n" }' > tmp
sed -e "1igene_name\t$(ls SPICE_*ReadsPerGene.out.tab | tr '\n' '\t' | sed 's/_ReadsPerGene.out.tab//g')" tmp > raw_counts_piceifrons_matrix.txt
```

## STAR and RSEM


## STAR and Salmon 


## HISAT2  


