---
title: "Sequence data quality control"
author: "Maeva Techer"
date: "`r Sys.Date()`"
output:
  workflowr::wflow_html
---

Load R libraries (install first from CRAN or Bioconductor)
```{r libraries, message=FALSE, warning=FALSE}
library("knitr")
library("rmdformats")
library("tidyverse")
library("DT")  # for making interactive search table
library("plotly") # for interactive plots
library("ggthemes") # for theme_calc
library("reshape2")

## Global options
options(max.print="10000")
knitr::opts_chunk$set(
	echo = TRUE,
	message = FALSE,
	warning = FALSE,
	cache = FALSE,
	comment = FALSE,
	prompt = FALSE,
	tidy = TRUE
)
opts_knit$set(width=75)
```

# Control the quality of fastq files

From the point we have the renamed, and paired-end / single-end read for the species of interest (one folder), we are ready to run our Snakemake pipeline on it. If not provided beforehand by TxGen, we need to check the quality of the sequences downloaded or generated using `FASTQC`. 
  
Results can be view by opening the *.html files in web browser.  

[WRITE ABOUT MULTIQC]


# Trim and adapter removal
After checking the initial sequence quality, we can determine any parameters change for `Trimmomatic`.

```
########################################
# Snakefile rule
########################################

rule trim_adapt:
	input:
		read1 = OUTdir + "/reads/{locust}_1.fastq.gz",
		read2 = OUTdir + "/reads/{locust}_2.fastq.gz",
		adaptfile = OUTdir + "/list/TruSeqNextera_PE.fa"
	output:
		trimmedread1 = OUTdir + "/trimming/{locust}_trim1P_1.fastq.gz",
		badread1 = OUTdir + "/trimming/{locust}_trim1U_1.fastq.gz",
		trimmedread2 = OUTdir + "/trimming/{locust}_trim2P_2.fastq.gz",
		badread2 = OUTdir + "/trimming/{locust}_trim2U_2.fastq.gz",
	shell:
		"""
		module load Trimmomatic/0.39-Java-11
		java -jar $EBROOTTRIMMOMATIC/trimmomatic-0.39.jar PE -threads 2 -phred33 {input.read1} {input.read2} {output.trimmedread1} {output.badread1} {output.trimmedread2} {output.badread2} ILLUMINACLIP:{input.adaptfile}:2:30:10 LEADING:30 TRAILING:30 SLIDINGWINDOW:4:15 MINLEN:36 
		"""
		
########################################
# Parameters in the cluster.json file
########################################

    "trim_adapt":
    {
        "cpus-per-task" : 2,
        "partition" : "medium",
	"ntasks": 2,
        "mem" : "1G",
        "time": "0-04:00:00"
    },

```

# Trim quality control

We always QC after trimming to ensure fine-tuning that the sequences clipping and filtering were not unnecessarily harsh. Given the number of sequences we work with, we do not need to go through each file immediately for time purposes. Instead, sample randomly across species, rearing conditions, and tissues to see that the process worked well. 
  
```
########################################
# Snakefile rule
########################################

#Quality control step after trimming: checked for adapter content in particular and quality scores
rule trim_fastqc:
	input:
		read1 = OUTdir + "/trimming/{locust}_trim1P_1.fastq.gz",
		read2 = OUTdir + "/trimming/{locust}_trim2P_2.fastq.gz",
	output:
		htmlqc1 = OUTdir + "/trimming/{locust}_trim1P_1_fastqc.html",
		htmlqc2 = OUTdir + "/trimming/{locust}_trim2P_2_fastqc.html",
	shell:
		"""
		module load FastQC/0.11.9-Java-11
		fastqc {input.read1}
		fastqc {input.read2}
		"""
		
########################################
# Parameters in the cluster.json file
########################################

    "trim_fastqc":
    {
        "cpus-per-task" : 2,
        "partition" : "medium",
        "ntasks": 1,
        "mem" : "500M",
        "time": "0-03:00:00"
    },

```
  
  
<figure class="center">
**EXAMPLE OF READS QUALITY BEFORE TRIMMING**  
<img src="img/FASTQC_before.png" alt="module_spider" width="800px"/>  
**EXAMPLE OF READS QUALITY AFTER TRIMMING**   
We can see that the sequence length has changed and that the 5' and 3' end positions with lower quality have been removed.  
<img src="img/FASTQC_after.png" alt="module_spider" width="800px"/>  
  
   
   
**EXAMPLE OF READS QUALITY BEFORE TRIMMING**   
<img src="img/FASTQC_before2.png" alt="module_spider" width="800px"/>  
**EXAMPLE OF READS QUALITY AFTER TRIMMING**   
We can see that most detected adapter sequences have been adequately removed after trimming.  
<img src="img/FASTQC_after2.png" alt="module_spider" width="800px"/>  
</figure>
  

# Check for sequencing data contamination  

Contamination is likely bound to happen during experiments, tissue acquisition, RNA extraction and library preparation. One can hope to reduces as much as possible its impact on the final sequencing data which can affect the mapping rate success.  

## Screen for microbial contamination

We decided to screen for microbes sequences present in the trimmed paired end reads fastq.gz files using the tool [`Kaiju`](https://kaiju.binf.ku.dk/). `Kaiju` translates metatranscriptomics sequencing reads into six possible reading frames and searches for maximum exact matches of amino acid sequences in a given annotation protein database.   

We used the most extensive microbial database `nr_euk` which encompass the subset of NCBI BLAST nr database containing all proteins belonging to Archaea, Bacteria, Viruses, Fungi and microbial Eukaryotes.

```
## Downloading the 2022-03-10 database from Kaiju webserver
wget https://kaiju.binf.ku.dk/database/kaiju_db_nr_euk_2022-03-10.tgz
```

To run `Kaiju` we used the following `Snakemake` rule:

```
########################################
# Snakefile rule
########################################

rule kaiju:
        input:
                read1 = OUTdir + "/trimming/{locust}_trim1P_R1_001.fastq.gz",
                read2 = OUTdir + "/trimming/{locust}_trim2P_R2_001.fastq.gz",
                database = KAIJUdir + "/kaiju_db_nr_euk.fmi",
                taxonid = KAIJUdir + "/nodes.dmp",
        output:
                report = OUTdir + "/kaiju/{locust}_kaiju.out",
        shell:
                """
                ml GCC/8.3.0  OpenMPI/3.1.4 Kaiju/1.7.3-Python-3.7.4
                kaiju -z 12 -v -a greedy -f {input.database} -t {input.taxonid} -i {input.read1} -j {input.read2} -o {output.report}
                """
                
		
########################################
# Parameters in the cluster.json file
########################################

    "kaiju":
    {
        "cpus-per-task" : 6,
        "partition" : "medium",
        "ntasks": 2,
        "mem" : "200G",
        "time": "0-12:00:00"
    },

```

## Visualize the metatranscriptomics result

`Kaiju` output can be exported to be view in a interactive and hierarchical multi-layered pie-charts using [`Krona`](https://github.com/marbl/Krona/wiki). The results are generated by a .html page. We followed the [Kaiju tutorial](https://github.com/bioinformatics-centre/kaiju) on the Github page:  

```
## Need to have Kaiju loaded
ml GCC/8.3.0  OpenMPI/3.1.4 Kaiju/1.7.3-Python-3.7.4
kaiju2krona -t nodes.dmp -n names.dmp -i {locust}_kaiju.out -o {locust}_kaiju.out.krona

## Need to use Krona functions
ml GCCcore/8.2.0 KronaTools/2.7.1
ktImportText -o {locust}_kaiju.out.html {locust}_kaiju.out.krona
```

## Filter for rRNA
