---
title: "Downloading or obtaining sequencing data"
author: "Maeva Techer"
date: "`r Sys.Date()`"
output:
  workflowr::wflow_html
---

### Load R libraries (install first from CRAN or Bioconductor)
```{r libraries, message=FALSE, warning=FALSE}
library("knitr")
library("rmdformats")
library("tidyverse")
library("DT")  # for making interactive search table
library("plotly") # for interactive plots
library("ggthemes") # for theme_calc
library("reshape2")

## Global options
options(max.print="10000")
knitr::opts_chunk$set(
	echo = TRUE,
	message = FALSE,
	warning = FALSE,
	cache = FALSE,
	comment = FALSE,
	prompt = FALSE,
	tidy = TRUE
)
opts_knit$set(width=75)
```

# Schistocerca reference genome

The BPRI generated high-quality and chromosome length genome for six species of grasshoppers and locusts _Schistocerca_. For this, the genome assembly team used a combination of long-reads sequencing with HiFi PacBio and short-reads with Hi-C. Following the finalization of the genomes assemblies, the annotation was conducted by the Eukaryotic Annotation Pipeline for RefSeq by NCBI.  

We will use the assemblies that received accession numbers for RefSeq and associated with .gtf and .gff files.

Below is an example for downloading _Schistocerca piceifrons_ from NCBI RefSeq:
```
## Downloading the genome sequence, primary assembly fasta file with RefSeq contig accessions
wget https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/021/461/385/GCF_021461385.2_iqSchPice1.1/GCF_021461385.2_iqSchPice1.1_genomic.fna.gz

## Downloading the comprehensive annotation file in .gtf format
wget https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/021/461/385/GCF_021461385.2_iqSchPice1.1/GCF_021461385.2_iqSchPice1.1_genomic.gtf.gz
```

# Obtaining RNA sequence reads

The reads that can be mapped on these sequences can be either from sequences with accession on NCBI, DDBJ or yet to be release after being generated _de novo_.  


## Downloading NCBI Transcriptome SRAs

We searched through the NCBI database for RNA SRA associated with _Schistocerca_ in general and found [**160 accessions**](https://www.ncbi.nlm.nih.gov/Traces/study/?query_key=2&WebEnv=MCID_62fb21f526f8fb0d3100cc06&o=acc_s%3Aa) were available. All RNA-seq data for _S. americana_, _S. piceifrons_, _S. cubense_, _S. cancellata_, _S. nitens_ were generated originally in the Song Lab at Texas A&M University and were produced following the same standard protocol.  

Using the Run Selector from NCBI, we can easily download a metadata table which we can use to visualize how the accessions are distributed per species. We will use this metadata table for analysis in which we include pre-generated SRA.  
  
We collect a list of accessions from *Run* for each species and then use `SRA-toolkit` from NCBI. First, we make an empty directory named `ncbi` to download each SRA. This is where `SRA Toolkit` will dump the prefetched SRA files in `.sra` format.

```
ml purge
ml GCC/10.2.0 OpenMPI/4.0.5 SRA-Toolkit/2.10.9
vdb-config --interactive
```
Once in the vdb-config interactive mode, select `cache`, `choose`, then use `[ .. ]`, to enter `/home/USERNAME/PATH/ncbi` one directory at a time

```
prefetch --option-file SraAccList.txt
cat SraAccList.txt | xargs fasterq-dump --split-3 --outdir "/your-directory/for-fastq"
```
Clean-up the `ncbi` directory and move the fastq.gz file (rename if wanted).

another option 
```
for x in *.sra ; do fasterq-dump --split-files $x ; mv *.fastq ../../paired_end_piceifrons/; done
```
  
## _De novo_ sequencing

We generated new whole transcriptomes from two density conditions for the desert locust (Acrididae: _Schistocerca gregaria_). Here, we analyze 20 transcriptomes for a pilot project using Illumina Stranded Total RNA with RiboZero depletion and sequenced on a NovaSeq SP flow cell at TxGen for a targeted amount of 40 millions reads per library.   

The details of the sequences are as follow:  
**Bulk tissue from 2nd generation solitary control**  
| Sample name                        | Description                                 |
| ---------------------------------- | ------------------------------------------- |
|GREG-HATCH-S1-FULL_S20| Solitary hatch-ling 1st instar from Pearl and Atticus   |	
|GREG-S-ICT-9-ALB_S6| Antenna lobes from replicate #9  	|
|GREG-S-ICT-9-ANT_S5| Antennae from replicate #9  |
|GREG-S-ICT-9-FAT_S3| Fat body from replicate #9  |
|GREG-S-ICT-9-MHB_S7| Mushroom body from replicate #9  |
|GREG-S-ICT-9-MOP_S4| Maxillary palps from replicate #9  |
|GREG-S-ICT-9-MTG_S2| Metathoracic ganglia from replicate #9  |
|GREG-S-ICT-9-OLB_S8| Optical lobes from replicate #9  |
|GREG-S-ICT-9-WG_S1| Whole gut from replicate #9  |
  
**Bulk tissue from highly crowded control**  
| Sample name                        | Description                                 |
| ---------------------------------- | ------------------------------------------- |
GREG-G-CCT-11-ALB_S14| Antenna lobes from replicate #11  |
GREG-G-CCT-11-ALB-FULL_S17| Antenna lobes from replicate #11  |
GREG-G-CCT-11-ANT_S13| Antennae from replicate #11  |
GREG-G-CCT-11-FAT_S11| Fat body from replicate #11  |
GREG-G-CCT-11-FAT-FULL_S19| Fat body from replicate #11  |
GREG-G-CCT-11-MHB_S15| Mushroom body from replicate #11  |
GREG-G-CCT-11-MOP_S12| Maxillary palps from replicate #11  |
GREG-G-CCT-11-MTG_S10| Metathoracic ganglia from replicate #11  |
GREG-G-CCT-11-OLB_S16| Optical lobes from replicate #11  |
GREG-G-CCT-11-OLB-FULL_S18| Optical lobes from replicate #11  |
GREG-G-CCT-11-WG_S9| Whole gut from replicate #11  |

## Merging several runs into a single fastq.gz file

For transcriptome that have yet to be released, it is common for libraries to be run on multiples lanes depending on the amount of reads needed. I used the following loop to merge several files with different Lanes:  

```
#!/bin/bash
# add the print $3 for TxGen reads as it is for samples S
for i in $(ls -1 *R1*.gz | awk -F '_' '{print $1"_"$2"_"$3}' | sort | uniq)
do echo $i
echo "Merging R1 ${i}"
cat "$i"_L00*_R1_001.fastq.gz > "$i"_MERGE_R1_001.fastq.gz
echo "Merging R2 ${i}"
cat "$i"_L00*_R2_001.fastq.gz > "$i"_MERGE_R2_001.fastq.gz
done;
```

# Preparing the metadata file

Make a `.csv` file with as much information as possible per sample/file name (e.g., Sample_ID, Species, Sex, RearingCondition). An interactive and searchable table is found below and even be downloaded directly.  

_NB: Throughout our analysis, we will complete this metadata file by adding other stats related to sequencing and mapping._ 

``` {r Metadata}

# Load our SRA metadata table
metaseq <- read_table("data/metadata/RNAseq_modified_METADATA2022.txt", col_names  = TRUE,  guess_max = 5000)

## Create an interactive search table
metaseq %>%
    datatable(extensions = "Buttons", options = list(dom = "Blfrtip", buttons = c("copy",
        "csv", "excel"), lengthMenu = list(c(10, 20, 50, 100, 200, -1), c(10, 20,
        50, 100, 200, "All"))))
```
